// jquery/dist/jquery.js
/*!
 * jQuery JavaScript Library v2.1.4
 * http://jquery.com/
 *
 * Includes Sizzle.js
 * http://sizzlejs.com/
 *
 * Copyright 2005, 2014 jQuery Foundation, Inc. and other contributors
 * Released under the MIT license
 * http://jquery.org/license
 *
 * Date: 2015-04-28T16:01Z
 */

(function( global, factory ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 0);

	if ( typeof module === "object" && typeof module.exports === "object" ) {
		// For CommonJS and CommonJS-like environments where a proper `window`
		// is present, execute the factory and get jQuery.
		// For environments that do not have a `window` with a `document`
		// (such as Node.js), expose a factory as module.exports.
		// This accentuates the need for the creation of a real `window`.
		// e.g. var jQuery = require("jquery")(window);
		// See ticket #14549 for more info.
		module.exports = global.document ?
			factory( global, true ) :
			function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 1);};
	} else {
		factory( global );
	}

// Pass this if window is not defined yet
}(typeof window !== "undefined" ? window : this, function( window, noGlobal ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 2);

// Support: Firefox 18+
// Can't be in strict mode, several libs including ASP.NET trace
// the stack via arguments.caller.callee and Firefox dies if
// you try to trace through "use strict" call chains. (#13335)
//

var arr = [];

var slice = arr.slice;

var concat = arr.concat;

var push = arr.push;

var indexOf = arr.indexOf;

var class2type = {};

var toString = class2type.toString;

var hasOwn = class2type.hasOwnProperty;

var support = {};



var
	// Use the correct document accordingly with window argument (sandbox)
	document = window.document,

	version = "2.1.4",

	// Define a local copy of jQuery
	jQuery = function( selector, context ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 3);
		// The jQuery object is actually just the init constructor 'enhanced'
		// Need init if jQuery is called (just allow error to be thrown if not included)
		return new jQuery.fn.init( selector, context );
	},

	// Support: Android<4.1
	// Make sure we trim BOM and NBSP
	rtrim = /^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g,

	// Matches dashed string for camelizing
	rmsPrefix = /^-ms-/,
	rdashAlpha = /-([\da-z])/gi,

	// Used by jQuery.camelCase as callback to replace()
	fcamelCase = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 4);};

jQuery.fn = jQuery.prototype = {
	// The current version of jQuery being used
	jquery: version,

	constructor: jQuery,

	// Start with an empty selector
	selector: "",

	// The default length of a jQuery object is 0
	length: 0,

	toArray: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 5);},

	// Get the Nth element in the matched element set OR
	// Get the whole matched element set as a clean array
	get: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 6);},

	// Take an array of elements and push it onto the stack
	// (returning the new matched element set)
	pushStack: function( elems ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 7);

		// Build a new jQuery matched element set
		var ret = jQuery.merge( this.constructor(), elems );

		// Add the old object onto the stack (as a reference)
		ret.prevObject = this;
		ret.context = this.context;

		// Return the newly-formed element set
		return ret;
	},

	// Execute a callback for every element in the matched set.
	// (You can seed the arguments with an array of args, but this is
	// only used internally.)
	each: function( callback, args ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 8);
		return jQuery.each( this, callback, args );
	},

	map: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 9);},

	slice: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 10);},

	first: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 11);},

	last: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 12);},

	eq: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 13);},

	end: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 14);},

	// For internal use only.
	// Behaves like an Array's method, not like a jQuery method.
	push: push,
	sort: arr.sort,
	splice: arr.splice
};

jQuery.extend = jQuery.fn.extend = function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 15);
	var options, name, src, copy, copyIsArray, clone,
		target = arguments[0] || {},
		i = 1,
		length = arguments.length,
		deep = false;

	// Handle a deep copy situation
	if ( typeof target === "boolean" ) {
		deep = target;

		// Skip the boolean and the target
		target = arguments[ i ] || {};
		i++;
	}

	// Handle case when target is a string or something (possible in deep copy)
	if ( typeof target !== "object" && !jQuery.isFunction(target) ) {
		target = {};
	}

	// Extend jQuery itself if only one argument is passed
	if ( i === length ) {
		target = this;
		i--;
	}

	for ( ; i < length; i++ ) {
		// Only deal with non-null/undefined values
		if ( (options = arguments[ i ]) != null ) {
			// Extend the base object
			for ( name in options ) {
				src = target[ name ];
				copy = options[ name ];

				// Prevent never-ending loop
				if ( target === copy ) {
					continue;
				}

				// Recurse if we're merging plain objects or arrays
				if ( deep && copy && ( jQuery.isPlainObject(copy) || (copyIsArray = jQuery.isArray(copy)) ) ) {
					if ( copyIsArray ) {
						copyIsArray = false;
						clone = src && jQuery.isArray(src) ? src : [];

					} else {
						clone = src && jQuery.isPlainObject(src) ? src : {};
					}

					// Never move original objects, clone them
					target[ name ] = jQuery.extend( deep, clone, copy );

				// Don't bring in undefined values
				} else if ( copy !== undefined ) {
					target[ name ] = copy;
				}
			}
		}
	}

	// Return the modified object
	return target;
};

jQuery.extend({
	// Unique for each copy of jQuery on the page
	expando: "jQuery" + ( version + Math.random() ).replace( /\D/g, "" ),

	// Assume jQuery is ready without the ready module
	isReady: true,

	error: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 16);},

	noop: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 17);},

	isFunction: function( obj ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 18);
		return jQuery.type(obj) === "function";
	},

	isArray: Array.isArray,

	isWindow: function( obj ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 19);
		return obj != null && obj === obj.window;
	},

	isNumeric: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 20);},

	isPlainObject: function( obj ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 21);
		// Not plain objects:
		// - Any object or value whose internal [[Class]] property is not "[object Object]"
		// - DOM nodes
		// - window
		if ( jQuery.type( obj ) !== "object" || obj.nodeType || jQuery.isWindow( obj ) ) {
			return false;
		}

		if ( obj.constructor &&
				!hasOwn.call( obj.constructor.prototype, "isPrototypeOf" ) ) {
			return false;
		}

		// If the function hasn't returned already, we're confident that
		// |obj| is a plain object, created by {} or constructed with new Object
		return true;
	},

	isEmptyObject: function( obj ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 22);
		var name;
		for ( name in obj ) {
			return false;
		}
		return true;
	},

	type: function( obj ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 23);
		if ( obj == null ) {
			return obj + "";
		}
		// Support: Android<4.0, iOS<6 (functionish RegExp)
		return typeof obj === "object" || typeof obj === "function" ?
			class2type[ toString.call(obj) ] || "object" :
			typeof obj;
	},

	// Evaluates a script in a global context
	globalEval: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 24);},

	// Convert dashed to camelCase; used by the css and data modules
	// Support: IE9-11+
	// Microsoft forgot to hump their vendor prefix (#9572)
	camelCase: function( string ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 25);
		return string.replace( rmsPrefix, "ms-" ).replace( rdashAlpha, fcamelCase );
	},

	nodeName: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 26);},

	// args is for internal usage only
	each: function( obj, callback, args ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 27);
		var value,
			i = 0,
			length = obj.length,
			isArray = isArraylike( obj );

		if ( args ) {
			if ( isArray ) {
				for ( ; i < length; i++ ) {
					value = callback.apply( obj[ i ], args );

					if ( value === false ) {
						break;
					}
				}
			} else {
				for ( i in obj ) {
					value = callback.apply( obj[ i ], args );

					if ( value === false ) {
						break;
					}
				}
			}

		// A special, fast, case for the most common use of each
		} else {
			if ( isArray ) {
				for ( ; i < length; i++ ) {
					value = callback.call( obj[ i ], i, obj[ i ] );

					if ( value === false ) {
						break;
					}
				}
			} else {
				for ( i in obj ) {
					value = callback.call( obj[ i ], i, obj[ i ] );

					if ( value === false ) {
						break;
					}
				}
			}
		}

		return obj;
	},

	// Support: Android<4.1
	trim: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 28);},

	// results is for internal usage only
	makeArray: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 29);},

	inArray: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 30);},

	merge: function( first, second ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 31);
		var len = +second.length,
			j = 0,
			i = first.length;

		for ( ; j < len; j++ ) {
			first[ i++ ] = second[ j ];
		}

		first.length = i;

		return first;
	},

	grep: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 32);},

	// arg is for internal usage only
	map: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 33);},

	// A global GUID counter for objects
	guid: 1,

	// Bind a function to a context, optionally partially applying any
	// arguments.
	proxy: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 34);},

	now: Date.now,

	// jQuery.support is not used in Core but other projects attach their
	// properties to it so it needs to exist.
	support: support
});

// Populate the class2type map
jQuery.each("Boolean Number String Function Array Date RegExp Object Error".split(" "), function(i, name) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 35);
	class2type[ "[object " + name + "]" ] = name.toLowerCase();
});

function isArraylike( obj ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 36);

	// Support: iOS 8.2 (not reproducible in simulator)
	// `in` check used to prevent JIT error (gh-2145)
	// hasOwn isn't used here due to false negatives
	// regarding Nodelist length in IE
	var length = "length" in obj && obj.length,
		type = jQuery.type( obj );

	if ( type === "function" || jQuery.isWindow( obj ) ) {
		return false;
	}

	if ( obj.nodeType === 1 && length ) {
		return true;
	}

	return type === "array" || length === 0 ||
		typeof length === "number" && length > 0 && ( length - 1 ) in obj;
}
var Sizzle =
/*!
 * Sizzle CSS Selector Engine v2.2.0-pre
 * http://sizzlejs.com/
 *
 * Copyright 2008, 2014 jQuery Foundation, Inc. and other contributors
 * Released under the MIT license
 * http://jquery.org/license
 *
 * Date: 2014-12-16
 */
(function( window ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 37);

var i,
	support,
	Expr,
	getText,
	isXML,
	tokenize,
	compile,
	select,
	outermostContext,
	sortInput,
	hasDuplicate,

	// Local document vars
	setDocument,
	document,
	docElem,
	documentIsHTML,
	rbuggyQSA,
	rbuggyMatches,
	matches,
	contains,

	// Instance-specific data
	expando = "sizzle" + 1 * new Date(),
	preferredDoc = window.document,
	dirruns = 0,
	done = 0,
	classCache = createCache(),
	tokenCache = createCache(),
	compilerCache = createCache(),
	sortOrder = function( a, b ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 38);
		if ( a === b ) {
			hasDuplicate = true;
		}
		return 0;
	},

	// General-purpose constants
	MAX_NEGATIVE = 1 << 31,

	// Instance methods
	hasOwn = ({}).hasOwnProperty,
	arr = [],
	pop = arr.pop,
	push_native = arr.push,
	push = arr.push,
	slice = arr.slice,
	// Use a stripped-down indexOf as it's faster than native
	// http://jsperf.com/thor-indexof-vs-for/5
	indexOf = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 39);},

	booleans = "checked|selected|async|autofocus|autoplay|controls|defer|disabled|hidden|ismap|loop|multiple|open|readonly|required|scoped",

	// Regular expressions

	// Whitespace characters http://www.w3.org/TR/css3-selectors/#whitespace
	whitespace = "[\\x20\\t\\r\\n\\f]",
	// http://www.w3.org/TR/css3-syntax/#characters
	characterEncoding = "(?:\\\\.|[\\w-]|[^\\x00-\\xa0])+",

	// Loosely modeled on CSS identifier characters
	// An unquoted value should be a CSS identifier http://www.w3.org/TR/css3-selectors/#attribute-selectors
	// Proper syntax: http://www.w3.org/TR/CSS21/syndata.html#value-def-identifier
	identifier = characterEncoding.replace( "w", "w#" ),

	// Attribute selectors: http://www.w3.org/TR/selectors/#attribute-selectors
	attributes = "\\[" + whitespace + "*(" + characterEncoding + ")(?:" + whitespace +
		// Operator (capture 2)
		"*([*^$|!~]?=)" + whitespace +
		// "Attribute values must be CSS identifiers [capture 5] or strings [capture 3 or capture 4]"
		"*(?:'((?:\\\\.|[^\\\\'])*)'|\"((?:\\\\.|[^\\\\\"])*)\"|(" + identifier + "))|)" + whitespace +
		"*\\]",

	pseudos = ":(" + characterEncoding + ")(?:\\((" +
		// To reduce the number of selectors needing tokenize in the preFilter, prefer arguments:
		// 1. quoted (capture 3; capture 4 or capture 5)
		"('((?:\\\\.|[^\\\\'])*)'|\"((?:\\\\.|[^\\\\\"])*)\")|" +
		// 2. simple (capture 6)
		"((?:\\\\.|[^\\\\()[\\]]|" + attributes + ")*)|" +
		// 3. anything else (capture 2)
		".*" +
		")\\)|)",

	// Leading and non-escaped trailing whitespace, capturing some non-whitespace characters preceding the latter
	rwhitespace = new RegExp( whitespace + "+", "g" ),
	rtrim = new RegExp( "^" + whitespace + "+|((?:^|[^\\\\])(?:\\\\.)*)" + whitespace + "+$", "g" ),

	rcomma = new RegExp( "^" + whitespace + "*," + whitespace + "*" ),
	rcombinators = new RegExp( "^" + whitespace + "*([>+~]|" + whitespace + ")" + whitespace + "*" ),

	rattributeQuotes = new RegExp( "=" + whitespace + "*([^\\]'\"]*?)" + whitespace + "*\\]", "g" ),

	rpseudo = new RegExp( pseudos ),
	ridentifier = new RegExp( "^" + identifier + "$" ),

	matchExpr = {
		"ID": new RegExp( "^#(" + characterEncoding + ")" ),
		"CLASS": new RegExp( "^\\.(" + characterEncoding + ")" ),
		"TAG": new RegExp( "^(" + characterEncoding.replace( "w", "w*" ) + ")" ),
		"ATTR": new RegExp( "^" + attributes ),
		"PSEUDO": new RegExp( "^" + pseudos ),
		"CHILD": new RegExp( "^:(only|first|last|nth|nth-last)-(child|of-type)(?:\\(" + whitespace +
			"*(even|odd|(([+-]|)(\\d*)n|)" + whitespace + "*(?:([+-]|)" + whitespace +
			"*(\\d+)|))" + whitespace + "*\\)|)", "i" ),
		"bool": new RegExp( "^(?:" + booleans + ")$", "i" ),
		// For use in libraries implementing .is()
		// We use this for POS matching in `select`
		"needsContext": new RegExp( "^" + whitespace + "*[>+~]|:(even|odd|eq|gt|lt|nth|first|last)(?:\\(" +
			whitespace + "*((?:-\\d)?\\d*)" + whitespace + "*\\)|)(?=[^-]|$)", "i" )
	},

	rinputs = /^(?:input|select|textarea|button)$/i,
	rheader = /^h\d$/i,

	rnative = /^[^{]+\{\s*\[native \w/,

	// Easily-parseable/retrievable ID or TAG or CLASS selectors
	rquickExpr = /^(?:#([\w-]+)|(\w+)|\.([\w-]+))$/,

	rsibling = /[+~]/,
	rescape = /'|\\/g,

	// CSS escapes http://www.w3.org/TR/CSS21/syndata.html#escaped-characters
	runescape = new RegExp( "\\\\([\\da-f]{1,6}" + whitespace + "?|(" + whitespace + ")|.)", "ig" ),
	funescape = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 40);},

	// Used for iframes
	// See setDocument()
	// Removing the function wrapper causes a "Permission Denied"
	// error in IE
	unloadHandler = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 41);};

// Optimize for push.apply( _, NodeList )
try {
	push.apply(
		(arr = slice.call( preferredDoc.childNodes )),
		preferredDoc.childNodes
	);
	// Support: Android<4.0
	// Detect silently failing push.apply
	arr[ preferredDoc.childNodes.length ].nodeType;
} catch ( e ) {
	push = { apply: arr.length ?

		// Leverage slice if possible
		function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 42);} :

		// Support: IE<9
		// Otherwise append directly
		function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 43);}
	};
}

function Sizzle( selector, context, results, seed ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 44);
	var match, elem, m, nodeType,
		// QSA vars
		i, groups, old, nid, newContext, newSelector;

	if ( ( context ? context.ownerDocument || context : preferredDoc ) !== document ) {
		setDocument( context );
	}

	context = context || document;
	results = results || [];
	nodeType = context.nodeType;

	if ( typeof selector !== "string" || !selector ||
		nodeType !== 1 && nodeType !== 9 && nodeType !== 11 ) {

		return results;
	}

	if ( !seed && documentIsHTML ) {

		// Try to shortcut find operations when possible (e.g., not under DocumentFragment)
		if ( nodeType !== 11 && (match = rquickExpr.exec( selector )) ) {
			// Speed-up: Sizzle("#ID")
			if ( (m = match[1]) ) {
				if ( nodeType === 9 ) {
					elem = context.getElementById( m );
					// Check parentNode to catch when Blackberry 4.6 returns
					// nodes that are no longer in the document (jQuery #6963)
					if ( elem && elem.parentNode ) {
						// Handle the case where IE, Opera, and Webkit return items
						// by name instead of ID
						if ( elem.id === m ) {
							results.push( elem );
							return results;
						}
					} else {
						return results;
					}
				} else {
					// Context is not a document
					if ( context.ownerDocument && (elem = context.ownerDocument.getElementById( m )) &&
						contains( context, elem ) && elem.id === m ) {
						results.push( elem );
						return results;
					}
				}

			// Speed-up: Sizzle("TAG")
			} else if ( match[2] ) {
				push.apply( results, context.getElementsByTagName( selector ) );
				return results;

			// Speed-up: Sizzle(".CLASS")
			} else if ( (m = match[3]) && support.getElementsByClassName ) {
				push.apply( results, context.getElementsByClassName( m ) );
				return results;
			}
		}

		// QSA path
		if ( support.qsa && (!rbuggyQSA || !rbuggyQSA.test( selector )) ) {
			nid = old = expando;
			newContext = context;
			newSelector = nodeType !== 1 && selector;

			// qSA works strangely on Element-rooted queries
			// We can work around this by specifying an extra ID on the root
			// and working up from there (Thanks to Andrew Dupont for the technique)
			// IE 8 doesn't work on object elements
			if ( nodeType === 1 && context.nodeName.toLowerCase() !== "object" ) {
				groups = tokenize( selector );

				if ( (old = context.getAttribute("id")) ) {
					nid = old.replace( rescape, "\\$&" );
				} else {
					context.setAttribute( "id", nid );
				}
				nid = "[id='" + nid + "'] ";

				i = groups.length;
				while ( i-- ) {
					groups[i] = nid + toSelector( groups[i] );
				}
				newContext = rsibling.test( selector ) && testContext( context.parentNode ) || context;
				newSelector = groups.join(",");
			}

			if ( newSelector ) {
				try {
					push.apply( results,
						newContext.querySelectorAll( newSelector )
					);
					return results;
				} catch(qsaError) {
				} finally {
					if ( !old ) {
						context.removeAttribute("id");
					}
				}
			}
		}
	}

	// All others
	return select( selector.replace( rtrim, "$1" ), context, results, seed );
}

/**
 * Create key-value caches of limited size
 * @returns {Function(string, Object)} Returns the Object data after storing it on itself with
 *	property name the (space-suffixed) string and (if the cache is larger than Expr.cacheLength)
 *	deleting the oldest entry
 */
function createCache() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 45);
	var keys = [];

	function cache(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 46);}
	return cache;
}

/**
 * Mark a function for special use by Sizzle
 * @param {Function} fn The function to mark
 */
function markFunction( fn ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 47);
	fn[ expando ] = true;
	return fn;
}

/**
 * Support testing using an element
 * @param {Function} fn Passed the created div and expects a boolean result
 */
function assert( fn ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 48);
	var div = document.createElement("div");

	try {
		return !!fn( div );
	} catch (e) {
		return false;
	} finally {
		// Remove from its parent by default
		if ( div.parentNode ) {
			div.parentNode.removeChild( div );
		}
		// release memory in IE
		div = null;
	}
}

/**
 * Adds the same handler for all of the specified attrs
 * @param {String} attrs Pipe-separated list of attributes
 * @param {Function} handler The method that will be applied
 */
function addHandle(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 49);}

/**
 * Checks document order of two siblings
 * @param {Element} a
 * @param {Element} b
 * @returns {Number} Returns less than 0 if a precedes b, greater than 0 if a follows b
 */
function siblingCheck(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 50);}

/**
 * Returns a function to use in pseudos for input types
 * @param {String} type
 */
function createInputPseudo( type ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 51);
	return function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 52);};
}

/**
 * Returns a function to use in pseudos for buttons
 * @param {String} type
 */
function createButtonPseudo( type ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 53);
	return function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 54);};
}

/**
 * Returns a function to use in pseudos for positionals
 * @param {Function} fn
 */
function createPositionalPseudo( fn ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 55);
	return markFunction(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 56);});
}

/**
 * Checks a node for validity as a Sizzle context
 * @param {Element|Object=} context
 * @returns {Element|Object|Boolean} The input node if acceptable, otherwise a falsy value
 */
function testContext(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 57);}

// Expose support vars for convenience
support = Sizzle.support = {};

/**
 * Detects XML nodes
 * @param {Element|Object} elem An element or a document
 * @returns {Boolean} True iff elem is a non-HTML XML node
 */
isXML = Sizzle.isXML = function( elem ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 58);
	// documentElement is verified for cases where it doesn't yet exist
	// (such as loading iframes in IE - #4833)
	var documentElement = elem && (elem.ownerDocument || elem).documentElement;
	return documentElement ? documentElement.nodeName !== "HTML" : false;
};

/**
 * Sets document-related variables once based on the current document
 * @param {Element|Object} [doc] An element or document object to use to set the document
 * @returns {Object} Returns the current document
 */
setDocument = Sizzle.setDocument = function( node ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 59);
	var hasCompare, parent,
		doc = node ? node.ownerDocument || node : preferredDoc;

	// If no document and documentElement is available, return
	if ( doc === document || doc.nodeType !== 9 || !doc.documentElement ) {
		return document;
	}

	// Set our document
	document = doc;
	docElem = doc.documentElement;
	parent = doc.defaultView;

	// Support: IE>8
	// If iframe document is assigned to "document" variable and if iframe has been reloaded,
	// IE will throw "permission denied" error when accessing "document" variable, see jQuery #13936
	// IE6-8 do not support the defaultView property so parent will be undefined
	if ( parent && parent !== parent.top ) {
		// IE11 does not have attachEvent, so all must suffer
		if ( parent.addEventListener ) {
			parent.addEventListener( "unload", unloadHandler, false );
		} else if ( parent.attachEvent ) {
			parent.attachEvent( "onunload", unloadHandler );
		}
	}

	/* Support tests
	---------------------------------------------------------------------- */
	documentIsHTML = !isXML( doc );

	/* Attributes
	---------------------------------------------------------------------- */

	// Support: IE<8
	// Verify that getAttribute really returns attributes and not properties
	// (excepting IE8 booleans)
	support.attributes = assert(function( div ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 60);
		div.className = "i";
		return !div.getAttribute("className");
	});

	/* getElement(s)By*
	---------------------------------------------------------------------- */

	// Check if getElementsByTagName("*") returns only elements
	support.getElementsByTagName = assert(function( div ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 61);
		div.appendChild( doc.createComment("") );
		return !div.getElementsByTagName("*").length;
	});

	// Support: IE<9
	support.getElementsByClassName = rnative.test( doc.getElementsByClassName );

	// Support: IE<10
	// Check if getElementById returns elements by name
	// The broken getElementById methods don't pick up programatically-set names,
	// so use a roundabout getElementsByName test
	support.getById = assert(function( div ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 62);
		docElem.appendChild( div ).id = expando;
		return !doc.getElementsByName || !doc.getElementsByName( expando ).length;
	});

	// ID find and filter
	if ( support.getById ) {
		Expr.find["ID"] = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 63);};
		Expr.filter["ID"] = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 64);};
	} else {
		// Support: IE6/7
		// getElementById is not reliable as a find shortcut
		delete Expr.find["ID"];

		Expr.filter["ID"] =  function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 65);};
	}

	// Tag
	Expr.find["TAG"] = support.getElementsByTagName ?
		function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 66);} :

		function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 67);};

	// Class
	Expr.find["CLASS"] = support.getElementsByClassName && function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 68);};

	/* QSA/matchesSelector
	---------------------------------------------------------------------- */

	// QSA and matchesSelector support

	// matchesSelector(:active) reports false when true (IE9/Opera 11.5)
	rbuggyMatches = [];

	// qSa(:focus) reports false when true (Chrome 21)
	// We allow this because of a bug in IE8/9 that throws an error
	// whenever `document.activeElement` is accessed on an iframe
	// So, we allow :focus to pass through QSA all the time to avoid the IE error
	// See http://bugs.jquery.com/ticket/13378
	rbuggyQSA = [];

	if ( (support.qsa = rnative.test( doc.querySelectorAll )) ) {
		// Build QSA regex
		// Regex strategy adopted from Diego Perini
		assert(function( div ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 69);
			// Select is set to empty string on purpose
			// This is to test IE's treatment of not explicitly
			// setting a boolean content attribute,
			// since its presence should be enough
			// http://bugs.jquery.com/ticket/12359
			docElem.appendChild( div ).innerHTML = "<a id='" + expando + "'></a>" +
				"<select id='" + expando + "-\f]' msallowcapture=''>" +
				"<option selected=''></option></select>";

			// Support: IE8, Opera 11-12.16
			// Nothing should be selected when empty strings follow ^= or $= or *=
			// The test attribute must be unknown in Opera but "safe" for WinRT
			// http://msdn.microsoft.com/en-us/library/ie/hh465388.aspx#attribute_section
			if ( div.querySelectorAll("[msallowcapture^='']").length ) {
				rbuggyQSA.push( "[*^$]=" + whitespace + "*(?:''|\"\")" );
			}

			// Support: IE8
			// Boolean attributes and "value" are not treated correctly
			if ( !div.querySelectorAll("[selected]").length ) {
				rbuggyQSA.push( "\\[" + whitespace + "*(?:value|" + booleans + ")" );
			}

			// Support: Chrome<29, Android<4.2+, Safari<7.0+, iOS<7.0+, PhantomJS<1.9.7+
			if ( !div.querySelectorAll( "[id~=" + expando + "-]" ).length ) {
				rbuggyQSA.push("~=");
			}

			// Webkit/Opera - :checked should return selected option elements
			// http://www.w3.org/TR/2011/REC-css3-selectors-20110929/#checked
			// IE8 throws error here and will not see later tests
			if ( !div.querySelectorAll(":checked").length ) {
				rbuggyQSA.push(":checked");
			}

			// Support: Safari 8+, iOS 8+
			// https://bugs.webkit.org/show_bug.cgi?id=136851
			// In-page `selector#id sibing-combinator selector` fails
			if ( !div.querySelectorAll( "a#" + expando + "+*" ).length ) {
				rbuggyQSA.push(".#.+[+~]");
			}
		});

		assert(function( div ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 70);
			// Support: Windows 8 Native Apps
			// The type and name attributes are restricted during .innerHTML assignment
			var input = doc.createElement("input");
			input.setAttribute( "type", "hidden" );
			div.appendChild( input ).setAttribute( "name", "D" );

			// Support: IE8
			// Enforce case-sensitivity of name attribute
			if ( div.querySelectorAll("[name=d]").length ) {
				rbuggyQSA.push( "name" + whitespace + "*[*^$|!~]?=" );
			}

			// FF 3.5 - :enabled/:disabled and hidden elements (hidden elements are still enabled)
			// IE8 throws error here and will not see later tests
			if ( !div.querySelectorAll(":enabled").length ) {
				rbuggyQSA.push( ":enabled", ":disabled" );
			}

			// Opera 10-11 does not throw on post-comma invalid pseudos
			div.querySelectorAll("*,:x");
			rbuggyQSA.push(",.*:");
		});
	}

	if ( (support.matchesSelector = rnative.test( (matches = docElem.matches ||
		docElem.webkitMatchesSelector ||
		docElem.mozMatchesSelector ||
		docElem.oMatchesSelector ||
		docElem.msMatchesSelector) )) ) {

		assert(function( div ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 71);
			// Check to see if it's possible to do matchesSelector
			// on a disconnected node (IE 9)
			support.disconnectedMatch = matches.call( div, "div" );

			// This should fail with an exception
			// Gecko does not error, returns false instead
			matches.call( div, "[s!='']:x" );
			rbuggyMatches.push( "!=", pseudos );
		});
	}

	rbuggyQSA = rbuggyQSA.length && new RegExp( rbuggyQSA.join("|") );
	rbuggyMatches = rbuggyMatches.length && new RegExp( rbuggyMatches.join("|") );

	/* Contains
	---------------------------------------------------------------------- */
	hasCompare = rnative.test( docElem.compareDocumentPosition );

	// Element contains another
	// Purposefully does not implement inclusive descendent
	// As in, an element does not contain itself
	contains = hasCompare || rnative.test( docElem.contains ) ?
		function( a, b ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 72);
			var adown = a.nodeType === 9 ? a.documentElement : a,
				bup = b && b.parentNode;
			return a === bup || !!( bup && bup.nodeType === 1 && (
				adown.contains ?
					adown.contains( bup ) :
					a.compareDocumentPosition && a.compareDocumentPosition( bup ) & 16
			));
		} :
		function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 73);};

	/* Sorting
	---------------------------------------------------------------------- */

	// Document order sorting
	sortOrder = hasCompare ?
	function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 74);} :
	function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 75);};

	return doc;
};

Sizzle.matches = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 76);};

Sizzle.matchesSelector = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 77);};

Sizzle.contains = function( context, elem ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 78);
	// Set document vars if needed
	if ( ( context.ownerDocument || context ) !== document ) {
		setDocument( context );
	}
	return contains( context, elem );
};

Sizzle.attr = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 79);};

Sizzle.error = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 80);};

/**
 * Document sorting and removing duplicates
 * @param {ArrayLike} results
 */
Sizzle.uniqueSort = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 81);};

/**
 * Utility function for retrieving the text value of an array of DOM nodes
 * @param {Array|Element} elem
 */
getText = Sizzle.getText = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 82);};

Expr = Sizzle.selectors = {

	// Can be adjusted by the user
	cacheLength: 50,

	createPseudo: markFunction,

	match: matchExpr,

	attrHandle: {},

	find: {},

	relative: {
		">": { dir: "parentNode", first: true },
		" ": { dir: "parentNode" },
		"+": { dir: "previousSibling", first: true },
		"~": { dir: "previousSibling" }
	},

	preFilter: {
		"ATTR": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 83);},

		"CHILD": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 84);},

		"PSEUDO": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 85);}
	},

	filter: {

		"TAG": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 86);},

		"CLASS": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 87);},

		"ATTR": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 88);},

		"CHILD": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 89);},

		"PSEUDO": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 90);}
	},

	pseudos: {
		// Potentially complex pseudos
		"not": markFunction(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 91);}),

		"has": markFunction(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 92);}),

		"contains": markFunction(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 93);}),

		// "Whether an element is represented by a :lang() selector
		// is based solely on the element's language value
		// being equal to the identifier C,
		// or beginning with the identifier C immediately followed by "-".
		// The matching of C against the element's language value is performed case-insensitively.
		// The identifier C does not have to be a valid language name."
		// http://www.w3.org/TR/selectors/#lang-pseudo
		"lang": markFunction( function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 94);}),

		// Miscellaneous
		"target": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 95);},

		"root": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 96);},

		"focus": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 97);},

		// Boolean properties
		"enabled": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 98);},

		"disabled": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 99);},

		"checked": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 100);},

		"selected": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 101);},

		// Contents
		"empty": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 102);},

		"parent": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 103);},

		// Element/input types
		"header": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 104);},

		"input": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 105);},

		"button": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 106);},

		"text": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 107);},

		// Position-in-collection
		"first": createPositionalPseudo(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 108);}),

		"last": createPositionalPseudo(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 109);}),

		"eq": createPositionalPseudo(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 110);}),

		"even": createPositionalPseudo(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 111);}),

		"odd": createPositionalPseudo(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 112);}),

		"lt": createPositionalPseudo(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 113);}),

		"gt": createPositionalPseudo(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 114);})
	}
};

Expr.pseudos["nth"] = Expr.pseudos["eq"];

// Add button/input type pseudos
for ( i in { radio: true, checkbox: true, file: true, password: true, image: true } ) {
	Expr.pseudos[ i ] = createInputPseudo( i );
}
for ( i in { submit: true, reset: true } ) {
	Expr.pseudos[ i ] = createButtonPseudo( i );
}

// Easy API for creating new setFilters
function setFilters() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 115);}
setFilters.prototype = Expr.filters = Expr.pseudos;
Expr.setFilters = new setFilters();

tokenize = Sizzle.tokenize = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 116);};

function toSelector(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 117);}

function addCombinator(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 118);}

function elementMatcher(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 119);}

function multipleContexts(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 120);}

function condense(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 121);}

function setMatcher(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 122);}

function matcherFromTokens(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 123);}

function matcherFromGroupMatchers(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 124);}

compile = Sizzle.compile = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 125);};

/**
 * A low-level selection function that works with Sizzle's compiled
 *  selector functions
 * @param {String|Function} selector A selector or a pre-compiled
 *  selector function built with Sizzle.compile
 * @param {Element} context
 * @param {Array} [results]
 * @param {Array} [seed] A set of elements to match against
 */
select = Sizzle.select = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 126);};

// One-time assignments

// Sort stability
support.sortStable = expando.split("").sort( sortOrder ).join("") === expando;

// Support: Chrome 14-35+
// Always assume duplicates if they aren't passed to the comparison function
support.detectDuplicates = !!hasDuplicate;

// Initialize against the default document
setDocument();

// Support: Webkit<537.32 - Safari 6.0.3/Chrome 25 (fixed in Chrome 27)
// Detached nodes confoundingly follow *each other*
support.sortDetached = assert(function( div1 ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 127);
	// Should return 1, but returns 4 (following)
	return div1.compareDocumentPosition( document.createElement("div") ) & 1;
});

// Support: IE<8
// Prevent attribute/property "interpolation"
// http://msdn.microsoft.com/en-us/library/ms536429%28VS.85%29.aspx
if ( !assert(function( div ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 128);
	div.innerHTML = "<a href='#'></a>";
	return div.firstChild.getAttribute("href") === "#" ;
}) ) {
	addHandle( "type|href|height|width", function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 129);});
}

// Support: IE<9
// Use defaultValue in place of getAttribute("value")
if ( !support.attributes || !assert(function( div ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 130);
	div.innerHTML = "<input/>";
	div.firstChild.setAttribute( "value", "" );
	return div.firstChild.getAttribute( "value" ) === "";
}) ) {
	addHandle( "value", function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 131);});
}

// Support: IE<9
// Use getAttributeNode to fetch booleans when getAttribute lies
if ( !assert(function( div ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 132);
	return div.getAttribute("disabled") == null;
}) ) {
	addHandle( booleans, function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 133);});
}

return Sizzle;

})( window );



jQuery.find = Sizzle;
jQuery.expr = Sizzle.selectors;
jQuery.expr[":"] = jQuery.expr.pseudos;
jQuery.unique = Sizzle.uniqueSort;
jQuery.text = Sizzle.getText;
jQuery.isXMLDoc = Sizzle.isXML;
jQuery.contains = Sizzle.contains;



var rneedsContext = jQuery.expr.match.needsContext;

var rsingleTag = (/^<(\w+)\s*\/?>(?:<\/\1>|)$/);



var risSimple = /^.[^:#\[\.,]*$/;

// Implement the identical functionality for filter and not
function winnow(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 134);}

jQuery.filter = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 135);};

jQuery.fn.extend({
	find: function( selector ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 136);
		var i,
			len = this.length,
			ret = [],
			self = this;

		if ( typeof selector !== "string" ) {
			return this.pushStack( jQuery( selector ).filter(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 137);}) );
		}

		for ( i = 0; i < len; i++ ) {
			jQuery.find( selector, self[ i ], ret );
		}

		// Needed because $( selector, context ) becomes $( context ).find( selector )
		ret = this.pushStack( len > 1 ? jQuery.unique( ret ) : ret );
		ret.selector = this.selector ? this.selector + " " + selector : selector;
		return ret;
	},
	filter: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 138);},
	not: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 139);},
	is: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 140);}
});


// Initialize a jQuery object


// A central reference to the root jQuery(document)
var rootjQuery,

	// A simple way to check for HTML strings
	// Prioritize #id over <tag> to avoid XSS via location.hash (#9521)
	// Strict HTML recognition (#11290: must start with <)
	rquickExpr = /^(?:\s*(<[\w\W]+>)[^>]*|#([\w-]*))$/,

	init = jQuery.fn.init = function( selector, context ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 141);
		var match, elem;

		// HANDLE: $(""), $(null), $(undefined), $(false)
		if ( !selector ) {
			return this;
		}

		// Handle HTML strings
		if ( typeof selector === "string" ) {
			if ( selector[0] === "<" && selector[ selector.length - 1 ] === ">" && selector.length >= 3 ) {
				// Assume that strings that start and end with <> are HTML and skip the regex check
				match = [ null, selector, null ];

			} else {
				match = rquickExpr.exec( selector );
			}

			// Match html or make sure no context is specified for #id
			if ( match && (match[1] || !context) ) {

				// HANDLE: $(html) -> $(array)
				if ( match[1] ) {
					context = context instanceof jQuery ? context[0] : context;

					// Option to run scripts is true for back-compat
					// Intentionally let the error be thrown if parseHTML is not present
					jQuery.merge( this, jQuery.parseHTML(
						match[1],
						context && context.nodeType ? context.ownerDocument || context : document,
						true
					) );

					// HANDLE: $(html, props)
					if ( rsingleTag.test( match[1] ) && jQuery.isPlainObject( context ) ) {
						for ( match in context ) {
							// Properties of context are called as methods if possible
							if ( jQuery.isFunction( this[ match ] ) ) {
								this[ match ]( context[ match ] );

							// ...and otherwise set as attributes
							} else {
								this.attr( match, context[ match ] );
							}
						}
					}

					return this;

				// HANDLE: $(#id)
				} else {
					elem = document.getElementById( match[2] );

					// Support: Blackberry 4.6
					// gEBID returns nodes no longer in the document (#6963)
					if ( elem && elem.parentNode ) {
						// Inject the element directly into the jQuery object
						this.length = 1;
						this[0] = elem;
					}

					this.context = document;
					this.selector = selector;
					return this;
				}

			// HANDLE: $(expr, $(...))
			} else if ( !context || context.jquery ) {
				return ( context || rootjQuery ).find( selector );

			// HANDLE: $(expr, context)
			// (which is just equivalent to: $(context).find(expr)
			} else {
				return this.constructor( context ).find( selector );
			}

		// HANDLE: $(DOMElement)
		} else if ( selector.nodeType ) {
			this.context = this[0] = selector;
			this.length = 1;
			return this;

		// HANDLE: $(function)
		// Shortcut for document ready
		} else if ( jQuery.isFunction( selector ) ) {
			return typeof rootjQuery.ready !== "undefined" ?
				rootjQuery.ready( selector ) :
				// Execute immediately if ready is not present
				selector( jQuery );
		}

		if ( selector.selector !== undefined ) {
			this.selector = selector.selector;
			this.context = selector.context;
		}

		return jQuery.makeArray( selector, this );
	};

// Give the init function the jQuery prototype for later instantiation
init.prototype = jQuery.fn;

// Initialize central reference
rootjQuery = jQuery( document );


var rparentsprev = /^(?:parents|prev(?:Until|All))/,
	// Methods guaranteed to produce a unique set when starting from a unique set
	guaranteedUnique = {
		children: true,
		contents: true,
		next: true,
		prev: true
	};

jQuery.extend({
	dir: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 142);},

	sibling: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 143);}
});

jQuery.fn.extend({
	has: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 144);},

	closest: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 145);},

	// Determine the position of an element within the set
	index: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 146);},

	add: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 147);},

	addBack: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 148);}
});

function sibling(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 149);}

jQuery.each({
	parent: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 150);},
	parents: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 151);},
	parentsUntil: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 152);},
	next: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 153);},
	prev: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 154);},
	nextAll: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 155);},
	prevAll: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 156);},
	nextUntil: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 157);},
	prevUntil: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 158);},
	siblings: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 159);},
	children: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 160);},
	contents: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 161);}
}, function( name, fn ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 162);
	jQuery.fn[ name ] = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 163);};
});
var rnotwhite = (/\S+/g);



// String to Object options format cache
var optionsCache = {};

// Convert String-formatted options into Object-formatted ones and store in cache
function createOptions( options ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 164);
	var object = optionsCache[ options ] = {};
	jQuery.each( options.match( rnotwhite ) || [], function( _, flag ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 165);
		object[ flag ] = true;
	});
	return object;
}

/*
 * Create a callback list using the following parameters:
 *
 *	options: an optional list of space-separated options that will change how
 *			the callback list behaves or a more traditional option object
 *
 * By default a callback list will act like an event callback list and can be
 * "fired" multiple times.
 *
 * Possible options:
 *
 *	once:			will ensure the callback list can only be fired once (like a Deferred)
 *
 *	memory:			will keep track of previous values and will call any callback added
 *					after the list has been fired right away with the latest "memorized"
 *					values (like a Deferred)
 *
 *	unique:			will ensure a callback can only be added once (no duplicate in the list)
 *
 *	stopOnFalse:	interrupt callings when a callback returns false
 *
 */
jQuery.Callbacks = function( options ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 166);

	// Convert options from String-formatted to Object-formatted if needed
	// (we check in cache first)
	options = typeof options === "string" ?
		( optionsCache[ options ] || createOptions( options ) ) :
		jQuery.extend( {}, options );

	var // Last fire value (for non-forgettable lists)
		memory,
		// Flag to know if list was already fired
		fired,
		// Flag to know if list is currently firing
		firing,
		// First callback to fire (used internally by add and fireWith)
		firingStart,
		// End of the loop when firing
		firingLength,
		// Index of currently firing callback (modified by remove if needed)
		firingIndex,
		// Actual callback list
		list = [],
		// Stack of fire calls for repeatable lists
		stack = !options.once && [],
		// Fire callbacks
		fire = function( data ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 167);
			memory = options.memory && data;
			fired = true;
			firingIndex = firingStart || 0;
			firingStart = 0;
			firingLength = list.length;
			firing = true;
			for ( ; list && firingIndex < firingLength; firingIndex++ ) {
				if ( list[ firingIndex ].apply( data[ 0 ], data[ 1 ] ) === false && options.stopOnFalse ) {
					memory = false; // To prevent further calls using add
					break;
				}
			}
			firing = false;
			if ( list ) {
				if ( stack ) {
					if ( stack.length ) {
						fire( stack.shift() );
					}
				} else if ( memory ) {
					list = [];
				} else {
					self.disable();
				}
			}
		},
		// Actual Callbacks object
		self = {
			// Add a callback or a collection of callbacks to the list
			add: function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 168);
				if ( list ) {
					// First, we save the current length
					var start = list.length;
					(function add( args ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 169);
						jQuery.each( args, function( _, arg ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 170);
							var type = jQuery.type( arg );
							if ( type === "function" ) {
								if ( !options.unique || !self.has( arg ) ) {
									list.push( arg );
								}
							} else if ( arg && arg.length && type !== "string" ) {
								// Inspect recursively
								add( arg );
							}
						});
					})( arguments );
					// Do we need to add the callbacks to the
					// current firing batch?
					if ( firing ) {
						firingLength = list.length;
					// With memory, if we're not firing then
					// we should call right away
					} else if ( memory ) {
						firingStart = start;
						fire( memory );
					}
				}
				return this;
			},
			// Remove a callback from the list
			remove: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 171);},
			// Check if a given callback is in the list.
			// If no argument is given, return whether or not list has callbacks attached.
			has: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 172);},
			// Remove all callbacks from the list
			empty: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 173);},
			// Have the list do nothing anymore
			disable: function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 174);
				list = stack = memory = undefined;
				return this;
			},
			// Is it disabled?
			disabled: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 175);},
			// Lock the list in its current state
			lock: function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 176);
				stack = undefined;
				if ( !memory ) {
					self.disable();
				}
				return this;
			},
			// Is it locked?
			locked: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 177);},
			// Call all callbacks with the given context and arguments
			fireWith: function( context, args ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 178);
				if ( list && ( !fired || stack ) ) {
					args = args || [];
					args = [ context, args.slice ? args.slice() : args ];
					if ( firing ) {
						stack.push( args );
					} else {
						fire( args );
					}
				}
				return this;
			},
			// Call all the callbacks with the given arguments
			fire: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 179);},
			// To know if the callbacks have already been called at least once
			fired: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 180);}
		};

	return self;
};


jQuery.extend({

	Deferred: function( func ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 181);
		var tuples = [
				// action, add listener, listener list, final state
				[ "resolve", "done", jQuery.Callbacks("once memory"), "resolved" ],
				[ "reject", "fail", jQuery.Callbacks("once memory"), "rejected" ],
				[ "notify", "progress", jQuery.Callbacks("memory") ]
			],
			state = "pending",
			promise = {
				state: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 182);},
				always: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 183);},
				then: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 184);},
				// Get a promise for this deferred
				// If obj is provided, the promise aspect is added to the object
				promise: function( obj ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 185);
					return obj != null ? jQuery.extend( obj, promise ) : promise;
				}
			},
			deferred = {};

		// Keep pipe for back-compat
		promise.pipe = promise.then;

		// Add list-specific methods
		jQuery.each( tuples, function( i, tuple ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 186);
			var list = tuple[ 2 ],
				stateString = tuple[ 3 ];

			// promise[ done | fail | progress ] = list.add
			promise[ tuple[1] ] = list.add;

			// Handle state
			if ( stateString ) {
				list.add(function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 187);
					// state = [ resolved | rejected ]
					state = stateString;

				// [ reject_list | resolve_list ].disable; progress_list.lock
				}, tuples[ i ^ 1 ][ 2 ].disable, tuples[ 2 ][ 2 ].lock );
			}

			// deferred[ resolve | reject | notify ]
			deferred[ tuple[0] ] = function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 188);
				deferred[ tuple[0] + "With" ]( this === deferred ? promise : this, arguments );
				return this;
			};
			deferred[ tuple[0] + "With" ] = list.fireWith;
		});

		// Make the deferred a promise
		promise.promise( deferred );

		// Call given func if any
		if ( func ) {
			func.call( deferred, deferred );
		}

		// All done!
		return deferred;
	},

	// Deferred helper
	when: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 189);}
});


// The deferred used on DOM ready
var readyList;

jQuery.fn.ready = function( fn ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 190);
	// Add the callback
	jQuery.ready.promise().done( fn );

	return this;
};

jQuery.extend({
	// Is the DOM ready to be used? Set to true once it occurs.
	isReady: false,

	// A counter to track how many items to wait for before
	// the ready event fires. See #6781
	readyWait: 1,

	// Hold (or release) the ready event
	holdReady: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 191);},

	// Handle when the DOM is ready
	ready: function( wait ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 192);

		// Abort if there are pending holds or we're already ready
		if ( wait === true ? --jQuery.readyWait : jQuery.isReady ) {
			return;
		}

		// Remember that the DOM is ready
		jQuery.isReady = true;

		// If a normal DOM Ready event fired, decrement, and wait if need be
		if ( wait !== true && --jQuery.readyWait > 0 ) {
			return;
		}

		// If there are functions bound, to execute
		readyList.resolveWith( document, [ jQuery ] );

		// Trigger any bound ready events
		if ( jQuery.fn.triggerHandler ) {
			jQuery( document ).triggerHandler( "ready" );
			jQuery( document ).off( "ready" );
		}
	}
});

/**
 * The ready event handler and self cleanup method
 */
function completed() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 193);
	document.removeEventListener( "DOMContentLoaded", completed, false );
	window.removeEventListener( "load", completed, false );
	jQuery.ready();
}

jQuery.ready.promise = function( obj ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 194);
	if ( !readyList ) {

		readyList = jQuery.Deferred();

		// Catch cases where $(document).ready() is called after the browser event has already occurred.
		// We once tried to use readyState "interactive" here, but it caused issues like the one
		// discovered by ChrisS here: http://bugs.jquery.com/ticket/12282#comment:15
		if ( document.readyState === "complete" ) {
			// Handle it asynchronously to allow scripts the opportunity to delay ready
			setTimeout( jQuery.ready );

		} else {

			// Use the handy event callback
			document.addEventListener( "DOMContentLoaded", completed, false );

			// A fallback to window.onload, that will always work
			window.addEventListener( "load", completed, false );
		}
	}
	return readyList.promise( obj );
};

// Kick off the DOM ready check even if the user does not
jQuery.ready.promise();




// Multifunctional method to get and set values of a collection
// The value/s can optionally be executed if it's a function
var access = jQuery.access = function( elems, fn, key, value, chainable, emptyGet, raw ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 195);
	var i = 0,
		len = elems.length,
		bulk = key == null;

	// Sets many values
	if ( jQuery.type( key ) === "object" ) {
		chainable = true;
		for ( i in key ) {
			jQuery.access( elems, fn, i, key[i], true, emptyGet, raw );
		}

	// Sets one value
	} else if ( value !== undefined ) {
		chainable = true;

		if ( !jQuery.isFunction( value ) ) {
			raw = true;
		}

		if ( bulk ) {
			// Bulk operations run against the entire set
			if ( raw ) {
				fn.call( elems, value );
				fn = null;

			// ...except when executing function values
			} else {
				bulk = fn;
				fn = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 196);};
			}
		}

		if ( fn ) {
			for ( ; i < len; i++ ) {
				fn( elems[i], key, raw ? value : value.call( elems[i], i, fn( elems[i], key ) ) );
			}
		}
	}

	return chainable ?
		elems :

		// Gets
		bulk ?
			fn.call( elems ) :
			len ? fn( elems[0], key ) : emptyGet;
};


/**
 * Determines whether an object can have data
 */
jQuery.acceptData = function( owner ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 197);
	// Accepts only:
	//  - Node
	//    - Node.ELEMENT_NODE
	//    - Node.DOCUMENT_NODE
	//  - Object
	//    - Any
	/* jshint -W018 */
	return owner.nodeType === 1 || owner.nodeType === 9 || !( +owner.nodeType );
};


function Data() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 198);
	// Support: Android<4,
	// Old WebKit does not have Object.preventExtensions/freeze method,
	// return new empty object instead with no [[set]] accessor
	Object.defineProperty( this.cache = {}, 0, {
		get: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 199);}
	});

	this.expando = jQuery.expando + Data.uid++;
}

Data.uid = 1;
Data.accepts = jQuery.acceptData;

Data.prototype = {
	key: function( owner ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 200);
		// We can accept data for non-element nodes in modern browsers,
		// but we should not, see #8335.
		// Always return the key for a frozen object.
		if ( !Data.accepts( owner ) ) {
			return 0;
		}

		var descriptor = {},
			// Check if the owner object already has a cache key
			unlock = owner[ this.expando ];

		// If not, create one
		if ( !unlock ) {
			unlock = Data.uid++;

			// Secure it in a non-enumerable, non-writable property
			try {
				descriptor[ this.expando ] = { value: unlock };
				Object.defineProperties( owner, descriptor );

			// Support: Android<4
			// Fallback to a less secure definition
			} catch ( e ) {
				descriptor[ this.expando ] = unlock;
				jQuery.extend( owner, descriptor );
			}
		}

		// Ensure the cache object
		if ( !this.cache[ unlock ] ) {
			this.cache[ unlock ] = {};
		}

		return unlock;
	},
	set: function( owner, data, value ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 201);
		var prop,
			// There may be an unlock assigned to this node,
			// if there is no entry for this "owner", create one inline
			// and set the unlock as though an owner entry had always existed
			unlock = this.key( owner ),
			cache = this.cache[ unlock ];

		// Handle: [ owner, key, value ] args
		if ( typeof data === "string" ) {
			cache[ data ] = value;

		// Handle: [ owner, { properties } ] args
		} else {
			// Fresh assignments by object are shallow copied
			if ( jQuery.isEmptyObject( cache ) ) {
				jQuery.extend( this.cache[ unlock ], data );
			// Otherwise, copy the properties one-by-one to the cache object
			} else {
				for ( prop in data ) {
					cache[ prop ] = data[ prop ];
				}
			}
		}
		return cache;
	},
	get: function( owner, key ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 202);
		// Either a valid cache is found, or will be created.
		// New caches will be created and the unlock returned,
		// allowing direct access to the newly created
		// empty data object. A valid owner object must be provided.
		var cache = this.cache[ this.key( owner ) ];

		return key === undefined ?
			cache : cache[ key ];
	},
	access: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 203);},
	remove: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 204);},
	hasData: function( owner ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 205);
		return !jQuery.isEmptyObject(
			this.cache[ owner[ this.expando ] ] || {}
		);
	},
	discard: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 206);}
};
var data_priv = new Data();

var data_user = new Data();



//	Implementation Summary
//
//	1. Enforce API surface and semantic compatibility with 1.9.x branch
//	2. Improve the module's maintainability by reducing the storage
//		paths to a single mechanism.
//	3. Use the same single mechanism to support "private" and "user" data.
//	4. _Never_ expose "private" data to user code (TODO: Drop _data, _removeData)
//	5. Avoid exposing implementation details on user objects (eg. expando properties)
//	6. Provide a clear path for implementation upgrade to WeakMap in 2014

var rbrace = /^(?:\{[\w\W]*\}|\[[\w\W]*\])$/,
	rmultiDash = /([A-Z])/g;

function dataAttr(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 207);}

jQuery.extend({
	hasData: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 208);},

	data: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 209);},

	removeData: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 210);},

	// TODO: Now that all calls to _data and _removeData have been replaced
	// with direct calls to data_priv methods, these can be deprecated.
	_data: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 211);},

	_removeData: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 212);}
});

jQuery.fn.extend({
	data: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 213);},

	removeData: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 214);}
});


jQuery.extend({
	queue: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 215);},

	dequeue: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 216);},

	// Not public - generate a queueHooks object, or return the current one
	_queueHooks: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 217);}
});

jQuery.fn.extend({
	queue: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 218);},
	dequeue: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 219);},
	clearQueue: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 220);},
	// Get a promise resolved when queues of a certain type
	// are emptied (fx is the type by default)
	promise: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 221);}
});
var pnum = (/[+-]?(?:\d*\.|)\d+(?:[eE][+-]?\d+|)/).source;

var cssExpand = [ "Top", "Right", "Bottom", "Left" ];

var isHidden = function( elem, el ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 222);
		// isHidden might be called from jQuery#filter function;
		// in that case, element will be second argument
		elem = el || elem;
		return jQuery.css( elem, "display" ) === "none" || !jQuery.contains( elem.ownerDocument, elem );
	};

var rcheckableType = (/^(?:checkbox|radio)$/i);



(function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 223);
	var fragment = document.createDocumentFragment(),
		div = fragment.appendChild( document.createElement( "div" ) ),
		input = document.createElement( "input" );

	// Support: Safari<=5.1
	// Check state lost if the name is set (#11217)
	// Support: Windows Web Apps (WWA)
	// `name` and `type` must use .setAttribute for WWA (#14901)
	input.setAttribute( "type", "radio" );
	input.setAttribute( "checked", "checked" );
	input.setAttribute( "name", "t" );

	div.appendChild( input );

	// Support: Safari<=5.1, Android<4.2
	// Older WebKit doesn't clone checked state correctly in fragments
	support.checkClone = div.cloneNode( true ).cloneNode( true ).lastChild.checked;

	// Support: IE<=11+
	// Make sure textarea (and checkbox) defaultValue is properly cloned
	div.innerHTML = "<textarea>x</textarea>";
	support.noCloneChecked = !!div.cloneNode( true ).lastChild.defaultValue;
})();
var strundefined = typeof undefined;



support.focusinBubbles = "onfocusin" in window;


var
	rkeyEvent = /^key/,
	rmouseEvent = /^(?:mouse|pointer|contextmenu)|click/,
	rfocusMorph = /^(?:focusinfocus|focusoutblur)$/,
	rtypenamespace = /^([^.]*)(?:\.(.+)|)$/;

function returnTrue(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 224);}

function returnFalse() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 225);
	return false;
}

function safeActiveElement(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 226);}

/*
 * Helper functions for managing events -- not part of the public interface.
 * Props to Dean Edwards' addEvent library for many of the ideas.
 */
jQuery.event = {

	global: {},

	add: function( elem, types, handler, data, selector ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 227);

		var handleObjIn, eventHandle, tmp,
			events, t, handleObj,
			special, handlers, type, namespaces, origType,
			elemData = data_priv.get( elem );

		// Don't attach events to noData or text/comment nodes (but allow plain objects)
		if ( !elemData ) {
			return;
		}

		// Caller can pass in an object of custom data in lieu of the handler
		if ( handler.handler ) {
			handleObjIn = handler;
			handler = handleObjIn.handler;
			selector = handleObjIn.selector;
		}

		// Make sure that the handler has a unique ID, used to find/remove it later
		if ( !handler.guid ) {
			handler.guid = jQuery.guid++;
		}

		// Init the element's event structure and main handler, if this is the first
		if ( !(events = elemData.events) ) {
			events = elemData.events = {};
		}
		if ( !(eventHandle = elemData.handle) ) {
			eventHandle = elemData.handle = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 228);};
		}

		// Handle multiple events separated by a space
		types = ( types || "" ).match( rnotwhite ) || [ "" ];
		t = types.length;
		while ( t-- ) {
			tmp = rtypenamespace.exec( types[t] ) || [];
			type = origType = tmp[1];
			namespaces = ( tmp[2] || "" ).split( "." ).sort();

			// There *must* be a type, no attaching namespace-only handlers
			if ( !type ) {
				continue;
			}

			// If event changes its type, use the special event handlers for the changed type
			special = jQuery.event.special[ type ] || {};

			// If selector defined, determine special event api type, otherwise given type
			type = ( selector ? special.delegateType : special.bindType ) || type;

			// Update special based on newly reset type
			special = jQuery.event.special[ type ] || {};

			// handleObj is passed to all event handlers
			handleObj = jQuery.extend({
				type: type,
				origType: origType,
				data: data,
				handler: handler,
				guid: handler.guid,
				selector: selector,
				needsContext: selector && jQuery.expr.match.needsContext.test( selector ),
				namespace: namespaces.join(".")
			}, handleObjIn );

			// Init the event handler queue if we're the first
			if ( !(handlers = events[ type ]) ) {
				handlers = events[ type ] = [];
				handlers.delegateCount = 0;

				// Only use addEventListener if the special events handler returns false
				if ( !special.setup || special.setup.call( elem, data, namespaces, eventHandle ) === false ) {
					if ( elem.addEventListener ) {
						elem.addEventListener( type, eventHandle, false );
					}
				}
			}

			if ( special.add ) {
				special.add.call( elem, handleObj );

				if ( !handleObj.handler.guid ) {
					handleObj.handler.guid = handler.guid;
				}
			}

			// Add to the element's handler list, delegates in front
			if ( selector ) {
				handlers.splice( handlers.delegateCount++, 0, handleObj );
			} else {
				handlers.push( handleObj );
			}

			// Keep track of which events have ever been used, for event optimization
			jQuery.event.global[ type ] = true;
		}

	},

	// Detach an event or set of events from an element
	remove: function( elem, types, handler, selector, mappedTypes ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 229);

		var j, origCount, tmp,
			events, t, handleObj,
			special, handlers, type, namespaces, origType,
			elemData = data_priv.hasData( elem ) && data_priv.get( elem );

		if ( !elemData || !(events = elemData.events) ) {
			return;
		}

		// Once for each type.namespace in types; type may be omitted
		types = ( types || "" ).match( rnotwhite ) || [ "" ];
		t = types.length;
		while ( t-- ) {
			tmp = rtypenamespace.exec( types[t] ) || [];
			type = origType = tmp[1];
			namespaces = ( tmp[2] || "" ).split( "." ).sort();

			// Unbind all events (on this namespace, if provided) for the element
			if ( !type ) {
				for ( type in events ) {
					jQuery.event.remove( elem, type + types[ t ], handler, selector, true );
				}
				continue;
			}

			special = jQuery.event.special[ type ] || {};
			type = ( selector ? special.delegateType : special.bindType ) || type;
			handlers = events[ type ] || [];
			tmp = tmp[2] && new RegExp( "(^|\\.)" + namespaces.join("\\.(?:.*\\.|)") + "(\\.|$)" );

			// Remove matching events
			origCount = j = handlers.length;
			while ( j-- ) {
				handleObj = handlers[ j ];

				if ( ( mappedTypes || origType === handleObj.origType ) &&
					( !handler || handler.guid === handleObj.guid ) &&
					( !tmp || tmp.test( handleObj.namespace ) ) &&
					( !selector || selector === handleObj.selector || selector === "**" && handleObj.selector ) ) {
					handlers.splice( j, 1 );

					if ( handleObj.selector ) {
						handlers.delegateCount--;
					}
					if ( special.remove ) {
						special.remove.call( elem, handleObj );
					}
				}
			}

			// Remove generic event handler if we removed something and no more handlers exist
			// (avoids potential for endless recursion during removal of special event handlers)
			if ( origCount && !handlers.length ) {
				if ( !special.teardown || special.teardown.call( elem, namespaces, elemData.handle ) === false ) {
					jQuery.removeEvent( elem, type, elemData.handle );
				}

				delete events[ type ];
			}
		}

		// Remove the expando if it's no longer used
		if ( jQuery.isEmptyObject( events ) ) {
			delete elemData.handle;
			data_priv.remove( elem, "events" );
		}
	},

	trigger: function( event, data, elem, onlyHandlers ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 230);

		var i, cur, tmp, bubbleType, ontype, handle, special,
			eventPath = [ elem || document ],
			type = hasOwn.call( event, "type" ) ? event.type : event,
			namespaces = hasOwn.call( event, "namespace" ) ? event.namespace.split(".") : [];

		cur = tmp = elem = elem || document;

		// Don't do events on text and comment nodes
		if ( elem.nodeType === 3 || elem.nodeType === 8 ) {
			return;
		}

		// focus/blur morphs to focusin/out; ensure we're not firing them right now
		if ( rfocusMorph.test( type + jQuery.event.triggered ) ) {
			return;
		}

		if ( type.indexOf(".") >= 0 ) {
			// Namespaced trigger; create a regexp to match event type in handle()
			namespaces = type.split(".");
			type = namespaces.shift();
			namespaces.sort();
		}
		ontype = type.indexOf(":") < 0 && "on" + type;

		// Caller can pass in a jQuery.Event object, Object, or just an event type string
		event = event[ jQuery.expando ] ?
			event :
			new jQuery.Event( type, typeof event === "object" && event );

		// Trigger bitmask: & 1 for native handlers; & 2 for jQuery (always true)
		event.isTrigger = onlyHandlers ? 2 : 3;
		event.namespace = namespaces.join(".");
		event.namespace_re = event.namespace ?
			new RegExp( "(^|\\.)" + namespaces.join("\\.(?:.*\\.|)") + "(\\.|$)" ) :
			null;

		// Clean up the event in case it is being reused
		event.result = undefined;
		if ( !event.target ) {
			event.target = elem;
		}

		// Clone any incoming data and prepend the event, creating the handler arg list
		data = data == null ?
			[ event ] :
			jQuery.makeArray( data, [ event ] );

		// Allow special events to draw outside the lines
		special = jQuery.event.special[ type ] || {};
		if ( !onlyHandlers && special.trigger && special.trigger.apply( elem, data ) === false ) {
			return;
		}

		// Determine event propagation path in advance, per W3C events spec (#9951)
		// Bubble up to document, then to window; watch for a global ownerDocument var (#9724)
		if ( !onlyHandlers && !special.noBubble && !jQuery.isWindow( elem ) ) {

			bubbleType = special.delegateType || type;
			if ( !rfocusMorph.test( bubbleType + type ) ) {
				cur = cur.parentNode;
			}
			for ( ; cur; cur = cur.parentNode ) {
				eventPath.push( cur );
				tmp = cur;
			}

			// Only add window if we got to document (e.g., not plain obj or detached DOM)
			if ( tmp === (elem.ownerDocument || document) ) {
				eventPath.push( tmp.defaultView || tmp.parentWindow || window );
			}
		}

		// Fire handlers on the event path
		i = 0;
		while ( (cur = eventPath[i++]) && !event.isPropagationStopped() ) {

			event.type = i > 1 ?
				bubbleType :
				special.bindType || type;

			// jQuery handler
			handle = ( data_priv.get( cur, "events" ) || {} )[ event.type ] && data_priv.get( cur, "handle" );
			if ( handle ) {
				handle.apply( cur, data );
			}

			// Native handler
			handle = ontype && cur[ ontype ];
			if ( handle && handle.apply && jQuery.acceptData( cur ) ) {
				event.result = handle.apply( cur, data );
				if ( event.result === false ) {
					event.preventDefault();
				}
			}
		}
		event.type = type;

		// If nobody prevented the default action, do it now
		if ( !onlyHandlers && !event.isDefaultPrevented() ) {

			if ( (!special._default || special._default.apply( eventPath.pop(), data ) === false) &&
				jQuery.acceptData( elem ) ) {

				// Call a native DOM method on the target with the same name name as the event.
				// Don't do default actions on window, that's where global variables be (#6170)
				if ( ontype && jQuery.isFunction( elem[ type ] ) && !jQuery.isWindow( elem ) ) {

					// Don't re-trigger an onFOO event when we call its FOO() method
					tmp = elem[ ontype ];

					if ( tmp ) {
						elem[ ontype ] = null;
					}

					// Prevent re-triggering of the same event, since we already bubbled it above
					jQuery.event.triggered = type;
					elem[ type ]();
					jQuery.event.triggered = undefined;

					if ( tmp ) {
						elem[ ontype ] = tmp;
					}
				}
			}
		}

		return event.result;
	},

	dispatch: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 231);},

	handlers: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 232);},

	// Includes some event props shared by KeyEvent and MouseEvent
	props: "altKey bubbles cancelable ctrlKey currentTarget eventPhase metaKey relatedTarget shiftKey target timeStamp view which".split(" "),

	fixHooks: {},

	keyHooks: {
		props: "char charCode key keyCode".split(" "),
		filter: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 233);}
	},

	mouseHooks: {
		props: "button buttons clientX clientY offsetX offsetY pageX pageY screenX screenY toElement".split(" "),
		filter: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 234);}
	},

	fix: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 235);},

	special: {
		load: {
			// Prevent triggered image.load events from bubbling to window.load
			noBubble: true
		},
		focus: {
			// Fire native event if possible so blur/focus sequence is correct
			trigger: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 236);},
			delegateType: "focusin"
		},
		blur: {
			trigger: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 237);},
			delegateType: "focusout"
		},
		click: {
			// For checkbox, fire native event so checked state will be right
			trigger: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 238);},

			// For cross-browser consistency, don't fire native .click() on links
			_default: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 239);}
		},

		beforeunload: {
			postDispatch: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 240);}
		}
	},

	simulate: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 241);}
};

jQuery.removeEvent = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 242);};

jQuery.Event = function( src, props ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 243);
	// Allow instantiation without the 'new' keyword
	if ( !(this instanceof jQuery.Event) ) {
		return new jQuery.Event( src, props );
	}

	// Event object
	if ( src && src.type ) {
		this.originalEvent = src;
		this.type = src.type;

		// Events bubbling up the document may have been marked as prevented
		// by a handler lower down the tree; reflect the correct value.
		this.isDefaultPrevented = src.defaultPrevented ||
				src.defaultPrevented === undefined &&
				// Support: Android<4.0
				src.returnValue === false ?
			returnTrue :
			returnFalse;

	// Event type
	} else {
		this.type = src;
	}

	// Put explicitly provided properties onto the event object
	if ( props ) {
		jQuery.extend( this, props );
	}

	// Create a timestamp if incoming event doesn't have one
	this.timeStamp = src && src.timeStamp || jQuery.now();

	// Mark it as fixed
	this[ jQuery.expando ] = true;
};

// jQuery.Event is based on DOM3 Events as specified by the ECMAScript Language Binding
// http://www.w3.org/TR/2003/WD-DOM-Level-3-Events-20030331/ecma-script-binding.html
jQuery.Event.prototype = {
	isDefaultPrevented: returnFalse,
	isPropagationStopped: returnFalse,
	isImmediatePropagationStopped: returnFalse,

	preventDefault: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 244);},
	stopPropagation: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 245);},
	stopImmediatePropagation: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 246);}
};

// Create mouseenter/leave events using mouseover/out and event-time checks
// Support: Chrome 15+
jQuery.each({
	mouseenter: "mouseover",
	mouseleave: "mouseout",
	pointerenter: "pointerover",
	pointerleave: "pointerout"
}, function( orig, fix ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 247);
	jQuery.event.special[ orig ] = {
		delegateType: fix,
		bindType: fix,

		handle: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 248);}
	};
});

// Support: Firefox, Chrome, Safari
// Create "bubbling" focus and blur events
if ( !support.focusinBubbles ) {
	jQuery.each({ focus: "focusin", blur: "focusout" }, function( orig, fix ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 249);

		// Attach a single capturing handler on the document while someone wants focusin/focusout
		var handler = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 250);};

		jQuery.event.special[ fix ] = {
			setup: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 251);},
			teardown: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 252);}
		};
	});
}

jQuery.fn.extend({

	on: function( types, selector, data, fn, /*INTERNAL*/ one ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 253);
		var origFn, type;

		// Types can be a map of types/handlers
		if ( typeof types === "object" ) {
			// ( types-Object, selector, data )
			if ( typeof selector !== "string" ) {
				// ( types-Object, data )
				data = data || selector;
				selector = undefined;
			}
			for ( type in types ) {
				this.on( type, selector, data, types[ type ], one );
			}
			return this;
		}

		if ( data == null && fn == null ) {
			// ( types, fn )
			fn = selector;
			data = selector = undefined;
		} else if ( fn == null ) {
			if ( typeof selector === "string" ) {
				// ( types, selector, fn )
				fn = data;
				data = undefined;
			} else {
				// ( types, data, fn )
				fn = data;
				data = selector;
				selector = undefined;
			}
		}
		if ( fn === false ) {
			fn = returnFalse;
		} else if ( !fn ) {
			return this;
		}

		if ( one === 1 ) {
			origFn = fn;
			fn = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 254);};
			// Use same guid so caller can remove using origFn
			fn.guid = origFn.guid || ( origFn.guid = jQuery.guid++ );
		}
		return this.each( function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 255);
			jQuery.event.add( this, types, fn, data, selector );
		});
	},
	one: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 256);},
	off: function( types, selector, fn ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 257);
		var handleObj, type;
		if ( types && types.preventDefault && types.handleObj ) {
			// ( event )  dispatched jQuery.Event
			handleObj = types.handleObj;
			jQuery( types.delegateTarget ).off(
				handleObj.namespace ? handleObj.origType + "." + handleObj.namespace : handleObj.origType,
				handleObj.selector,
				handleObj.handler
			);
			return this;
		}
		if ( typeof types === "object" ) {
			// ( types-object [, selector] )
			for ( type in types ) {
				this.off( type, selector, types[ type ] );
			}
			return this;
		}
		if ( selector === false || typeof selector === "function" ) {
			// ( types [, fn] )
			fn = selector;
			selector = undefined;
		}
		if ( fn === false ) {
			fn = returnFalse;
		}
		return this.each(function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 258);
			jQuery.event.remove( this, types, fn, selector );
		});
	},

	trigger: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 259);},
	triggerHandler: function( type, data ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 260);
		var elem = this[0];
		if ( elem ) {
			return jQuery.event.trigger( type, data, elem, true );
		}
	}
});


var
	rxhtmlTag = /<(?!area|br|col|embed|hr|img|input|link|meta|param)(([\w:]+)[^>]*)\/>/gi,
	rtagName = /<([\w:]+)/,
	rhtml = /<|&#?\w+;/,
	rnoInnerhtml = /<(?:script|style|link)/i,
	// checked="checked" or checked
	rchecked = /checked\s*(?:[^=]|=\s*.checked.)/i,
	rscriptType = /^$|\/(?:java|ecma)script/i,
	rscriptTypeMasked = /^true\/(.*)/,
	rcleanScript = /^\s*<!(?:\[CDATA\[|--)|(?:\]\]|--)>\s*$/g,

	// We have to close these tags to support XHTML (#13200)
	wrapMap = {

		// Support: IE9
		option: [ 1, "<select multiple='multiple'>", "</select>" ],

		thead: [ 1, "<table>", "</table>" ],
		col: [ 2, "<table><colgroup>", "</colgroup></table>" ],
		tr: [ 2, "<table><tbody>", "</tbody></table>" ],
		td: [ 3, "<table><tbody><tr>", "</tr></tbody></table>" ],

		_default: [ 0, "", "" ]
	};

// Support: IE9
wrapMap.optgroup = wrapMap.option;

wrapMap.tbody = wrapMap.tfoot = wrapMap.colgroup = wrapMap.caption = wrapMap.thead;
wrapMap.th = wrapMap.td;

// Support: 1.x compatibility
// Manipulating tables requires a tbody
function manipulationTarget(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 261);}

// Replace/restore the type attribute of script elements for safe DOM manipulation
function disableScript(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 262);}
function restoreScript(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 263);}

// Mark scripts as having already been evaluated
function setGlobalEval(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 264);}

function cloneCopyEvent(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 265);}

function getAll( context, tag ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 266);
	var ret = context.getElementsByTagName ? context.getElementsByTagName( tag || "*" ) :
			context.querySelectorAll ? context.querySelectorAll( tag || "*" ) :
			[];

	return tag === undefined || tag && jQuery.nodeName( context, tag ) ?
		jQuery.merge( [ context ], ret ) :
		ret;
}

// Fix IE bugs, see support tests
function fixInput(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 267);}

jQuery.extend({
	clone: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 268);},

	buildFragment: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 269);},

	cleanData: function( elems ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 270);
		var data, elem, type, key,
			special = jQuery.event.special,
			i = 0;

		for ( ; (elem = elems[ i ]) !== undefined; i++ ) {
			if ( jQuery.acceptData( elem ) ) {
				key = elem[ data_priv.expando ];

				if ( key && (data = data_priv.cache[ key ]) ) {
					if ( data.events ) {
						for ( type in data.events ) {
							if ( special[ type ] ) {
								jQuery.event.remove( elem, type );

							// This is a shortcut to avoid jQuery.event.remove's overhead
							} else {
								jQuery.removeEvent( elem, type, data.handle );
							}
						}
					}
					if ( data_priv.cache[ key ] ) {
						// Discard any remaining `private` data
						delete data_priv.cache[ key ];
					}
				}
			}
			// Discard any remaining `user` data
			delete data_user.cache[ elem[ data_user.expando ] ];
		}
	}
});

jQuery.fn.extend({
	text: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 271);},

	append: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 272);},

	prepend: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 273);},

	before: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 274);},

	after: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 275);},

	remove: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 276);},

	empty: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 277);},

	clone: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 278);},

	html: function( value ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 279);
		return access( this, function( value ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 280);
			var elem = this[ 0 ] || {},
				i = 0,
				l = this.length;

			if ( value === undefined && elem.nodeType === 1 ) {
				return elem.innerHTML;
			}

			// See if we can take a shortcut and just use innerHTML
			if ( typeof value === "string" && !rnoInnerhtml.test( value ) &&
				!wrapMap[ ( rtagName.exec( value ) || [ "", "" ] )[ 1 ].toLowerCase() ] ) {

				value = value.replace( rxhtmlTag, "<$1></$2>" );

				try {
					for ( ; i < l; i++ ) {
						elem = this[ i ] || {};

						// Remove element nodes and prevent memory leaks
						if ( elem.nodeType === 1 ) {
							jQuery.cleanData( getAll( elem, false ) );
							elem.innerHTML = value;
						}
					}

					elem = 0;

				// If using innerHTML throws an exception, use the fallback method
				} catch( e ) {}
			}

			if ( elem ) {
				this.empty().append( value );
			}
		}, null, value, arguments.length );
	},

	replaceWith: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 281);},

	detach: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 282);},

	domManip: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 283);}
});

jQuery.each({
	appendTo: "append",
	prependTo: "prepend",
	insertBefore: "before",
	insertAfter: "after",
	replaceAll: "replaceWith"
}, function( name, original ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 284);
	jQuery.fn[ name ] = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 285);};
});


var iframe,
	elemdisplay = {};

/**
 * Retrieve the actual display of a element
 * @param {String} name nodeName of the element
 * @param {Object} doc Document object
 */
// Called only from within defaultDisplay
function actualDisplay(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 286);}

/**
 * Try to determine the default display value of an element
 * @param {String} nodeName
 */
function defaultDisplay(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 287);}
var rmargin = (/^margin/);

var rnumnonpx = new RegExp( "^(" + pnum + ")(?!px)[a-z%]+$", "i" );

var getStyles = function( elem ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 288);
		// Support: IE<=11+, Firefox<=30+ (#15098, #14150)
		// IE throws on elements created in popups
		// FF meanwhile throws on frame elements through "defaultView.getComputedStyle"
		if ( elem.ownerDocument.defaultView.opener ) {
			return elem.ownerDocument.defaultView.getComputedStyle( elem, null );
		}

		return window.getComputedStyle( elem, null );
	};



function curCSS( elem, name, computed ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 289);
	var width, minWidth, maxWidth, ret,
		style = elem.style;

	computed = computed || getStyles( elem );

	// Support: IE9
	// getPropertyValue is only needed for .css('filter') (#12537)
	if ( computed ) {
		ret = computed.getPropertyValue( name ) || computed[ name ];
	}

	if ( computed ) {

		if ( ret === "" && !jQuery.contains( elem.ownerDocument, elem ) ) {
			ret = jQuery.style( elem, name );
		}

		// Support: iOS < 6
		// A tribute to the "awesome hack by Dean Edwards"
		// iOS < 6 (at least) returns percentage for a larger set of values, but width seems to be reliably pixels
		// this is against the CSSOM draft spec: http://dev.w3.org/csswg/cssom/#resolved-values
		if ( rnumnonpx.test( ret ) && rmargin.test( name ) ) {

			// Remember the original values
			width = style.width;
			minWidth = style.minWidth;
			maxWidth = style.maxWidth;

			// Put in the new values to get a computed value out
			style.minWidth = style.maxWidth = style.width = ret;
			ret = computed.width;

			// Revert the changed values
			style.width = width;
			style.minWidth = minWidth;
			style.maxWidth = maxWidth;
		}
	}

	return ret !== undefined ?
		// Support: IE
		// IE returns zIndex value as an integer.
		ret + "" :
		ret;
}


function addGetHookIf( conditionFn, hookFn ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 290);
	// Define the hook, we'll check on the first run if it's really needed.
	return {
		get: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 291);}
	};
}


(function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 292);
	var pixelPositionVal, boxSizingReliableVal,
		docElem = document.documentElement,
		container = document.createElement( "div" ),
		div = document.createElement( "div" );

	if ( !div.style ) {
		return;
	}

	// Support: IE9-11+
	// Style of cloned element affects source element cloned (#8908)
	div.style.backgroundClip = "content-box";
	div.cloneNode( true ).style.backgroundClip = "";
	support.clearCloneStyle = div.style.backgroundClip === "content-box";

	container.style.cssText = "border:0;width:0;height:0;top:0;left:-9999px;margin-top:1px;" +
		"position:absolute";
	container.appendChild( div );

	// Executing both pixelPosition & boxSizingReliable tests require only one layout
	// so they're executed at the same time to save the second computation.
	function computePixelPositionAndBoxSizingReliable(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 293);}

	// Support: node.js jsdom
	// Don't assume that getComputedStyle is a property of the global object
	if ( window.getComputedStyle ) {
		jQuery.extend( support, {
			pixelPosition: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 294);},
			boxSizingReliable: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 295);},
			reliableMarginRight: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 296);}
		});
	}
})();


// A method for quickly swapping in/out CSS properties to get correct calculations.
jQuery.swap = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 297);};


var
	// Swappable if display is none or starts with table except "table", "table-cell", or "table-caption"
	// See here for display values: https://developer.mozilla.org/en-US/docs/CSS/display
	rdisplayswap = /^(none|table(?!-c[ea]).+)/,
	rnumsplit = new RegExp( "^(" + pnum + ")(.*)$", "i" ),
	rrelNum = new RegExp( "^([+-])=(" + pnum + ")", "i" ),

	cssShow = { position: "absolute", visibility: "hidden", display: "block" },
	cssNormalTransform = {
		letterSpacing: "0",
		fontWeight: "400"
	},

	cssPrefixes = [ "Webkit", "O", "Moz", "ms" ];

// Return a css property mapped to a potentially vendor prefixed property
function vendorPropName( style, name ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 298);

	// Shortcut for names that are not vendor prefixed
	if ( name in style ) {
		return name;
	}

	// Check for vendor prefixed names
	var capName = name[0].toUpperCase() + name.slice(1),
		origName = name,
		i = cssPrefixes.length;

	while ( i-- ) {
		name = cssPrefixes[ i ] + capName;
		if ( name in style ) {
			return name;
		}
	}

	return origName;
}

function setPositiveNumber(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 299);}

function augmentWidthOrHeight(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 300);}

function getWidthOrHeight(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 301);}

function showHide( elements, show ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 302);
	var display, elem, hidden,
		values = [],
		index = 0,
		length = elements.length;

	for ( ; index < length; index++ ) {
		elem = elements[ index ];
		if ( !elem.style ) {
			continue;
		}

		values[ index ] = data_priv.get( elem, "olddisplay" );
		display = elem.style.display;
		if ( show ) {
			// Reset the inline display of this element to learn if it is
			// being hidden by cascaded rules or not
			if ( !values[ index ] && display === "none" ) {
				elem.style.display = "";
			}

			// Set elements which have been overridden with display: none
			// in a stylesheet to whatever the default browser style is
			// for such an element
			if ( elem.style.display === "" && isHidden( elem ) ) {
				values[ index ] = data_priv.access( elem, "olddisplay", defaultDisplay(elem.nodeName) );
			}
		} else {
			hidden = isHidden( elem );

			if ( display !== "none" || !hidden ) {
				data_priv.set( elem, "olddisplay", hidden ? display : jQuery.css( elem, "display" ) );
			}
		}
	}

	// Set the display of most of the elements in a second loop
	// to avoid the constant reflow
	for ( index = 0; index < length; index++ ) {
		elem = elements[ index ];
		if ( !elem.style ) {
			continue;
		}
		if ( !show || elem.style.display === "none" || elem.style.display === "" ) {
			elem.style.display = show ? values[ index ] || "" : "none";
		}
	}

	return elements;
}

jQuery.extend({

	// Add in style property hooks for overriding the default
	// behavior of getting and setting a style property
	cssHooks: {
		opacity: {
			get: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 303);}
		}
	},

	// Don't automatically add "px" to these possibly-unitless properties
	cssNumber: {
		"columnCount": true,
		"fillOpacity": true,
		"flexGrow": true,
		"flexShrink": true,
		"fontWeight": true,
		"lineHeight": true,
		"opacity": true,
		"order": true,
		"orphans": true,
		"widows": true,
		"zIndex": true,
		"zoom": true
	},

	// Add in properties whose names you wish to fix before
	// setting or getting the value
	cssProps: {
		"float": "cssFloat"
	},

	// Get and set the style property on a DOM Node
	style: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 304);},

	css: function( elem, name, extra, styles ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 305);
		var val, num, hooks,
			origName = jQuery.camelCase( name );

		// Make sure that we're working with the right name
		name = jQuery.cssProps[ origName ] || ( jQuery.cssProps[ origName ] = vendorPropName( elem.style, origName ) );

		// Try prefixed name followed by the unprefixed name
		hooks = jQuery.cssHooks[ name ] || jQuery.cssHooks[ origName ];

		// If a hook was provided get the computed value from there
		if ( hooks && "get" in hooks ) {
			val = hooks.get( elem, true, extra );
		}

		// Otherwise, if a way to get the computed value exists, use that
		if ( val === undefined ) {
			val = curCSS( elem, name, styles );
		}

		// Convert "normal" to computed value
		if ( val === "normal" && name in cssNormalTransform ) {
			val = cssNormalTransform[ name ];
		}

		// Make numeric if forced or a qualifier was provided and val looks numeric
		if ( extra === "" || extra ) {
			num = parseFloat( val );
			return extra === true || jQuery.isNumeric( num ) ? num || 0 : val;
		}
		return val;
	}
});

jQuery.each([ "height", "width" ], function( i, name ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 306);
	jQuery.cssHooks[ name ] = {
		get: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 307);},

		set: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 308);}
	};
});

// Support: Android 2.3
jQuery.cssHooks.marginRight = addGetHookIf( support.reliableMarginRight,
	function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 309);}
);

// These hooks are used by animate to expand properties
jQuery.each({
	margin: "",
	padding: "",
	border: "Width"
}, function( prefix, suffix ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 310);
	jQuery.cssHooks[ prefix + suffix ] = {
		expand: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 311);}
	};

	if ( !rmargin.test( prefix ) ) {
		jQuery.cssHooks[ prefix + suffix ].set = setPositiveNumber;
	}
});

jQuery.fn.extend({
	css: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 312);},
	show: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 313);},
	hide: function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 314);
		return showHide( this );
	},
	toggle: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 315);}
});


function Tween(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 316);}
jQuery.Tween = Tween;

Tween.prototype = {
	constructor: Tween,
	init: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 317);},
	cur: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 318);},
	run: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 319);}
};

Tween.prototype.init.prototype = Tween.prototype;

Tween.propHooks = {
	_default: {
		get: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 320);},
		set: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 321);}
	}
};

// Support: IE9
// Panic based approach to setting things on disconnected nodes
Tween.propHooks.scrollTop = Tween.propHooks.scrollLeft = {
	set: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 322);}
};

jQuery.easing = {
	linear: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 323);},
	swing: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 324);}
};

jQuery.fx = Tween.prototype.init;

// Back Compat <1.8 extension point
jQuery.fx.step = {};




var
	fxNow, timerId,
	rfxtypes = /^(?:toggle|show|hide)$/,
	rfxnum = new RegExp( "^(?:([+-])=|)(" + pnum + ")([a-z%]*)$", "i" ),
	rrun = /queueHooks$/,
	animationPrefilters = [ defaultPrefilter ],
	tweeners = {
		"*": [ function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 325);} ]
	};

// Animations created synchronously will run synchronously
function createFxNow(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 326);}

// Generate parameters to create a standard animation
function genFx( type, includeWidth ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 327);
	var which,
		i = 0,
		attrs = { height: type };

	// If we include width, step value is 1 to do all cssExpand values,
	// otherwise step value is 2 to skip over Left and Right
	includeWidth = includeWidth ? 1 : 0;
	for ( ; i < 4 ; i += 2 - includeWidth ) {
		which = cssExpand[ i ];
		attrs[ "margin" + which ] = attrs[ "padding" + which ] = type;
	}

	if ( includeWidth ) {
		attrs.opacity = attrs.width = type;
	}

	return attrs;
}

function createTween(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 328);}

function defaultPrefilter(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 329);}

function propFilter(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 330);}

function Animation(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 331);}

jQuery.Animation = jQuery.extend( Animation, {

	tweener: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 332);},

	prefilter: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 333);}
});

jQuery.speed = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 334);};

jQuery.fn.extend({
	fadeTo: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 335);},
	animate: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 336);},
	stop: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 337);},
	finish: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 338);}
});

jQuery.each([ "toggle", "show", "hide" ], function( i, name ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 339);
	var cssFn = jQuery.fn[ name ];
	jQuery.fn[ name ] = function( speed, easing, callback ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 340);
		return speed == null || typeof speed === "boolean" ?
			cssFn.apply( this, arguments ) :
			this.animate( genFx( name, true ), speed, easing, callback );
	};
});

// Generate shortcuts for custom animations
jQuery.each({
	slideDown: genFx("show"),
	slideUp: genFx("hide"),
	slideToggle: genFx("toggle"),
	fadeIn: { opacity: "show" },
	fadeOut: { opacity: "hide" },
	fadeToggle: { opacity: "toggle" }
}, function( name, props ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 341);
	jQuery.fn[ name ] = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 342);};
});

jQuery.timers = [];
jQuery.fx.tick = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 343);};

jQuery.fx.timer = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 344);};

jQuery.fx.interval = 13;

jQuery.fx.start = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 345);};

jQuery.fx.stop = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 346);};

jQuery.fx.speeds = {
	slow: 600,
	fast: 200,
	// Default speed
	_default: 400
};


// Based off of the plugin by Clint Helfers, with permission.
// http://blindsignals.com/index.php/2009/07/jquery-delay/
jQuery.fn.delay = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 347);};


(function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 348);
	var input = document.createElement( "input" ),
		select = document.createElement( "select" ),
		opt = select.appendChild( document.createElement( "option" ) );

	input.type = "checkbox";

	// Support: iOS<=5.1, Android<=4.2+
	// Default value for a checkbox should be "on"
	support.checkOn = input.value !== "";

	// Support: IE<=11+
	// Must access selectedIndex to make default options select
	support.optSelected = opt.selected;

	// Support: Android<=2.3
	// Options inside disabled selects are incorrectly marked as disabled
	select.disabled = true;
	support.optDisabled = !opt.disabled;

	// Support: IE<=11+
	// An input loses its value after becoming a radio
	input = document.createElement( "input" );
	input.value = "t";
	input.type = "radio";
	support.radioValue = input.value === "t";
})();


var nodeHook, boolHook,
	attrHandle = jQuery.expr.attrHandle;

jQuery.fn.extend({
	attr: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 349);},

	removeAttr: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 350);}
});

jQuery.extend({
	attr: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 351);},

	removeAttr: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 352);},

	attrHooks: {
		type: {
			set: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 353);}
		}
	}
});

// Hooks for boolean attributes
boolHook = {
	set: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 354);}
};
jQuery.each( jQuery.expr.match.bool.source.match( /\w+/g ), function( i, name ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 355);
	var getter = attrHandle[ name ] || jQuery.find.attr;

	attrHandle[ name ] = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 356);};
});




var rfocusable = /^(?:input|select|textarea|button)$/i;

jQuery.fn.extend({
	prop: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 357);},

	removeProp: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 358);}
});

jQuery.extend({
	propFix: {
		"for": "htmlFor",
		"class": "className"
	},

	prop: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 359);},

	propHooks: {
		tabIndex: {
			get: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 360);}
		}
	}
});

if ( !support.optSelected ) {
	jQuery.propHooks.selected = {
		get: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 361);}
	};
}

jQuery.each([
	"tabIndex",
	"readOnly",
	"maxLength",
	"cellSpacing",
	"cellPadding",
	"rowSpan",
	"colSpan",
	"useMap",
	"frameBorder",
	"contentEditable"
], function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 362);
	jQuery.propFix[ this.toLowerCase() ] = this;
});




var rclass = /[\t\r\n\f]/g;

jQuery.fn.extend({
	addClass: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 363);},

	removeClass: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 364);},

	toggleClass: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 365);},

	hasClass: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 366);}
});




var rreturn = /\r/g;

jQuery.fn.extend({
	val: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 367);}
});

jQuery.extend({
	valHooks: {
		option: {
			get: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 368);}
		},
		select: {
			get: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 369);},

			set: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 370);}
		}
	}
});

// Radios and checkboxes getter/setter
jQuery.each([ "radio", "checkbox" ], function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 371);
	jQuery.valHooks[ this ] = {
		set: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 372);}
	};
	if ( !support.checkOn ) {
		jQuery.valHooks[ this ].get = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 373);};
	}
});




// Return jQuery for attributes-only inclusion


jQuery.each( ("blur focus focusin focusout load resize scroll unload click dblclick " +
	"mousedown mouseup mousemove mouseover mouseout mouseenter mouseleave " +
	"change select submit keydown keypress keyup error contextmenu").split(" "), function( i, name ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 374);

	// Handle event binding
	jQuery.fn[ name ] = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 375);};
});

jQuery.fn.extend({
	hover: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 376);},

	bind: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 377);},
	unbind: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 378);},

	delegate: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 379);},
	undelegate: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 380);}
});


var nonce = jQuery.now();

var rquery = (/\?/);



// Support: Android 2.3
// Workaround failure to string-cast null input
jQuery.parseJSON = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 381);};


// Cross-browser xml parsing
jQuery.parseXML = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 382);};


var
	rhash = /#.*$/,
	rts = /([?&])_=[^&]*/,
	rheaders = /^(.*?):[ \t]*([^\r\n]*)$/mg,
	// #7653, #8125, #8152: local protocol detection
	rlocalProtocol = /^(?:about|app|app-storage|.+-extension|file|res|widget):$/,
	rnoContent = /^(?:GET|HEAD)$/,
	rprotocol = /^\/\//,
	rurl = /^([\w.+-]+:)(?:\/\/(?:[^\/?#]*@|)([^\/?#:]*)(?::(\d+)|)|)/,

	/* Prefilters
	 * 1) They are useful to introduce custom dataTypes (see ajax/jsonp.js for an example)
	 * 2) These are called:
	 *    - BEFORE asking for a transport
	 *    - AFTER param serialization (s.data is a string if s.processData is true)
	 * 3) key is the dataType
	 * 4) the catchall symbol "*" can be used
	 * 5) execution will start with transport dataType and THEN continue down to "*" if needed
	 */
	prefilters = {},

	/* Transports bindings
	 * 1) key is the dataType
	 * 2) the catchall symbol "*" can be used
	 * 3) selection will start with transport dataType and THEN go to "*" if needed
	 */
	transports = {},

	// Avoid comment-prolog char sequence (#10098); must appease lint and evade compression
	allTypes = "*/".concat( "*" ),

	// Document location
	ajaxLocation = window.location.href,

	// Segment location into parts
	ajaxLocParts = rurl.exec( ajaxLocation.toLowerCase() ) || [];

// Base "constructor" for jQuery.ajaxPrefilter and jQuery.ajaxTransport
function addToPrefiltersOrTransports( structure ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 383);

	// dataTypeExpression is optional and defaults to "*"
	return function( dataTypeExpression, func ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 384);

		if ( typeof dataTypeExpression !== "string" ) {
			func = dataTypeExpression;
			dataTypeExpression = "*";
		}

		var dataType,
			i = 0,
			dataTypes = dataTypeExpression.toLowerCase().match( rnotwhite ) || [];

		if ( jQuery.isFunction( func ) ) {
			// For each dataType in the dataTypeExpression
			while ( (dataType = dataTypes[i++]) ) {
				// Prepend if requested
				if ( dataType[0] === "+" ) {
					dataType = dataType.slice( 1 ) || "*";
					(structure[ dataType ] = structure[ dataType ] || []).unshift( func );

				// Otherwise append
				} else {
					(structure[ dataType ] = structure[ dataType ] || []).push( func );
				}
			}
		}
	};
}

// Base inspection function for prefilters and transports
function inspectPrefiltersOrTransports(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 385);}

// A special extend for ajax options
// that takes "flat" options (not to be deep extended)
// Fixes #9887
function ajaxExtend( target, src ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 386);
	var key, deep,
		flatOptions = jQuery.ajaxSettings.flatOptions || {};

	for ( key in src ) {
		if ( src[ key ] !== undefined ) {
			( flatOptions[ key ] ? target : ( deep || (deep = {}) ) )[ key ] = src[ key ];
		}
	}
	if ( deep ) {
		jQuery.extend( true, target, deep );
	}

	return target;
}

/* Handles responses to an ajax request:
 * - finds the right dataType (mediates between content-type and expected dataType)
 * - returns the corresponding response
 */
function ajaxHandleResponses(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 387);}

/* Chain conversions given the request and the original response
 * Also sets the responseXXX fields on the jqXHR instance
 */
function ajaxConvert(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 388);}

jQuery.extend({

	// Counter for holding the number of active queries
	active: 0,

	// Last-Modified header cache for next request
	lastModified: {},
	etag: {},

	ajaxSettings: {
		url: ajaxLocation,
		type: "GET",
		isLocal: rlocalProtocol.test( ajaxLocParts[ 1 ] ),
		global: true,
		processData: true,
		async: true,
		contentType: "application/x-www-form-urlencoded; charset=UTF-8",
		/*
		timeout: 0,
		data: null,
		dataType: null,
		username: null,
		password: null,
		cache: null,
		throws: false,
		traditional: false,
		headers: {},
		*/

		accepts: {
			"*": allTypes,
			text: "text/plain",
			html: "text/html",
			xml: "application/xml, text/xml",
			json: "application/json, text/javascript"
		},

		contents: {
			xml: /xml/,
			html: /html/,
			json: /json/
		},

		responseFields: {
			xml: "responseXML",
			text: "responseText",
			json: "responseJSON"
		},

		// Data converters
		// Keys separate source (or catchall "*") and destination types with a single space
		converters: {

			// Convert anything to text
			"* text": String,

			// Text to html (true = no transformation)
			"text html": true,

			// Evaluate text as a json expression
			"text json": jQuery.parseJSON,

			// Parse text as xml
			"text xml": jQuery.parseXML
		},

		// For options that shouldn't be deep extended:
		// you can add your own custom options here if
		// and when you create one that shouldn't be
		// deep extended (see ajaxExtend)
		flatOptions: {
			url: true,
			context: true
		}
	},

	// Creates a full fledged settings object into target
	// with both ajaxSettings and settings fields.
	// If target is omitted, writes into ajaxSettings.
	ajaxSetup: function( target, settings ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 389);
		return settings ?

			// Building a settings object
			ajaxExtend( ajaxExtend( target, jQuery.ajaxSettings ), settings ) :

			// Extending ajaxSettings
			ajaxExtend( jQuery.ajaxSettings, target );
	},

	ajaxPrefilter: addToPrefiltersOrTransports( prefilters ),
	ajaxTransport: addToPrefiltersOrTransports( transports ),

	// Main method
	ajax: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 390);},

	getJSON: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 391);},

	getScript: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 392);}
});

jQuery.each( [ "get", "post" ], function( i, method ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 393);
	jQuery[ method ] = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 394);};
});


jQuery._evalUrl = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 395);};


jQuery.fn.extend({
	wrapAll: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 396);},

	wrapInner: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 397);},

	wrap: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 398);},

	unwrap: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 399);}
});


jQuery.expr.filters.hidden = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 400);};
jQuery.expr.filters.visible = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 401);};




var r20 = /%20/g,
	rbracket = /\[\]$/,
	rCRLF = /\r?\n/g,
	rsubmitterTypes = /^(?:submit|button|image|reset|file)$/i,
	rsubmittable = /^(?:input|select|textarea|keygen)/i;

function buildParams(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 402);}

// Serialize an array of form elements or a set of
// key/values into a query string
jQuery.param = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 403);};

jQuery.fn.extend({
	serialize: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 404);},
	serializeArray: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 405);}
});


jQuery.ajaxSettings.xhr = function() {___jdce_logger("/node_modules/jquery/dist/jquery.js", 406);
	try {
		return new XMLHttpRequest();
	} catch( e ) {}
};

var xhrId = 0,
	xhrCallbacks = {},
	xhrSuccessStatus = {
		// file protocol always yields status code 0, assume 200
		0: 200,
		// Support: IE9
		// #1450: sometimes IE returns 1223 when it should be 204
		1223: 204
	},
	xhrSupported = jQuery.ajaxSettings.xhr();

// Support: IE9
// Open requests must be manually aborted on unload (#5280)
// See https://support.microsoft.com/kb/2856746 for more info
if ( window.attachEvent ) {
	window.attachEvent( "onunload", function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 407);});
}

support.cors = !!xhrSupported && ( "withCredentials" in xhrSupported );
support.ajax = xhrSupported = !!xhrSupported;

jQuery.ajaxTransport(function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 408);});




// Install script dataType
jQuery.ajaxSetup({
	accepts: {
		script: "text/javascript, application/javascript, application/ecmascript, application/x-ecmascript"
	},
	contents: {
		script: /(?:java|ecma)script/
	},
	converters: {
		"text script": function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 409);}
	}
});

// Handle cache's special case and crossDomain
jQuery.ajaxPrefilter( "script", function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 410);});

// Bind script tag hack transport
jQuery.ajaxTransport( "script", function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 411);});




var oldCallbacks = [],
	rjsonp = /(=)\?(?=&|$)|\?\?/;

// Default jsonp settings
jQuery.ajaxSetup({
	jsonp: "callback",
	jsonpCallback: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 412);}
});

// Detect, normalize options and install callbacks for jsonp requests
jQuery.ajaxPrefilter( "json jsonp", function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 413);});




// data: string of html
// context (optional): If specified, the fragment will be created in this context, defaults to document
// keepScripts (optional): If true, will include scripts passed in the html string
jQuery.parseHTML = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 414);};


// Keep a copy of the old load method
var _load = jQuery.fn.load;

/**
 * Load a url into a page
 */
jQuery.fn.load = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 415);};




// Attach a bunch of functions for handling common AJAX events
jQuery.each( [ "ajaxStart", "ajaxStop", "ajaxComplete", "ajaxError", "ajaxSuccess", "ajaxSend" ], function( i, type ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 416);
	jQuery.fn[ type ] = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 417);};
});




jQuery.expr.filters.animated = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 418);};




var docElem = window.document.documentElement;

/**
 * Gets a window from an element
 */
function getWindow(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 419);}

jQuery.offset = {
	setOffset: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 420);}
};

jQuery.fn.extend({
	offset: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 421);},

	position: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 422);},

	offsetParent: function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 423);}
});

// Create scrollLeft and scrollTop methods
jQuery.each( { scrollLeft: "pageXOffset", scrollTop: "pageYOffset" }, function( method, prop ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 424);
	var top = "pageYOffset" === prop;

	jQuery.fn[ method ] = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 425);};
});

// Support: Safari<7+, Chrome<37+
// Add the top/left cssHooks using jQuery.fn.position
// Webkit bug: https://bugs.webkit.org/show_bug.cgi?id=29084
// Blink bug: https://code.google.com/p/chromium/issues/detail?id=229280
// getComputedStyle returns percent when specified for top/left/bottom/right;
// rather than make the css module depend on the offset module, just check for it here
jQuery.each( [ "top", "left" ], function( i, prop ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 426);
	jQuery.cssHooks[ prop ] = addGetHookIf( support.pixelPosition,
		function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 427);}
	);
});


// Create innerHeight, innerWidth, height, width, outerHeight and outerWidth methods
jQuery.each( { Height: "height", Width: "width" }, function( name, type ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 428);
	jQuery.each( { padding: "inner" + name, content: type, "": "outer" + name }, function( defaultExtra, funcName ) {___jdce_logger("/node_modules/jquery/dist/jquery.js", 429);
		// Margin is only for outerHeight, outerWidth
		jQuery.fn[ funcName ] = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 430);};
	});
});


// The number of elements contained in the matched element set
jQuery.fn.size = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 431);};

jQuery.fn.andSelf = jQuery.fn.addBack;




// Register as a named AMD module, since jQuery can be concatenated with other
// files that may use define, but not via a proper concatenation script that
// understands anonymous AMD modules. A named AMD is safest and most robust
// way to register. Lowercase jquery is used because AMD module names are
// derived from file names, and jQuery is normally delivered in a lowercase
// file name. Do this after creating the global so that if an AMD module wants
// to call noConflict to hide this version of jQuery, it will work.

// Note that for maximum portability, libraries that are not jQuery should
// declare themselves as anonymous modules, and avoid setting a global if an
// AMD loader is present. jQuery is a special case. For more information, see
// https://github.com/jrburke/requirejs/wiki/Updating-existing-libraries#wiki-anon

if ( typeof define === "function" && define.amd ) {
	define( "jquery", [], function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 432);});
}




var
	// Map over jQuery in case of overwrite
	_jQuery = window.jQuery,

	// Map over the $ in case of overwrite
	_$ = window.$;

jQuery.noConflict = function(){___jdce_logger("/node_modules/jquery/dist/jquery.js", 433);};

// Expose jQuery and $ identifiers, even in AMD
// (#7102#comment:10, https://github.com/jquery/jquery/pull/557)
// and CommonJS for browser emulators (#13566)
if ( typeof noGlobal === strundefined ) {
	window.jQuery = window.$ = jQuery;
}




return jQuery;

}));
